Differential gene expression analysis of the gut of gnotobiotic honey
bees
================
Joanito Liberti, University of Lausanne

## Check data quality with FastQC

``` bash
for f in *.fastq.gz; do 
fastqc $f -o /work/FASTQC/; 
done
```

## Quality trimming with Trimmomatic

``` bash
# We use GNU parallel to run Trimmomatic on multiple fastq.gz files simultaneously
parallel 'java -jar /work/Software/Trimmomatic-0.39/trimmomatic-0.39.jar SE {} /work/07_Trimmomatic/{.}_trim.fastq.gz ILLUMINACLIP:/work/Software/Trimmomatic-0.39/adapters/TruSeq3-SE.fa:2:30:10 LEADING:10 TRAILING:10 SLIDINGWINDOW:4:20 MINLEN:80' ::: *.fastq.gz
```

## Map reads to the honey bee genome (Amel_HAv3.1) with STAR

``` bash
# Create genome indices, Amel_HAv3.1
STAR --runThreadN 10 --runMode genomeGenerate --genomeDir /work/02_genome/HAv3.1 --genomeFastaFiles /work/02_genome/HAv3.1/GCF_003254395.2_Amel_HAv3.1_genomic.fna --sjdbGTFfile /work/02_genome/HAv3.1/GCF_003254395.2_Amel_HAv3.1_genomic.gff --sjdbGTFfeatureExon exon --sjdbGTFtagExonParentTranscript Parent --sjdbOverhang 150 &

## Map Trimmomatic filtered fastq files against the Hav3.1 genome:
for f in *.fastq.gz
do
STAR --genomeDir /work/02_genome/HAv3.1/ --readFilesIn /work/07_Trimmomatic/Guts/$f --outFileNamePrefix /work/08_Alignments_HavGenome/Alignments_Guts_HAv3.1/$f --outSAMtype BAM SortedByCoordinate --limitBAMsortRAM 30000000000 --sjdbGTFfile /work/02_genome/HAv3.1/GCF_003254395.2_Amel_HAv3.1_genomic.gff --sjdbGTFfeatureExon exon --sjdbGTFtagExonParentTranscript transcript_id --readFilesCommand zcat --sjdbOverhang 150 --runThreadN 20
done
```

## As each sample was sequenced across two lanes we need to merge the .bam files

``` bash
printf '%s\0' *.bam | sort -zV | xargs -0 -n2 sh -c 'out="${1%_L*}"; echo samtools merge -@ 20 "${out}.bam" "$1" "$2"' sh # first check that the command would be fine, then remove echo. -@ 20 means use 20 threads
```

## Count features with HTseq-count

``` bash
time (for f in *.bam
do 
echo ${f}
htseq-count -f bam -r pos -s no -m union -t exon -i gene_id $f /work/02_genome/HAv3.1/GCF_003254395.2_Amel_HAv3.1_genomic.gtf > ${f/.bam/_raw_counts.txt}
done)

# Remove first lines as these are the total counts
for f in *txt
do
tail -n +2 $f > $f.tmp && mv $f.tmp $f
done
```

## Merge count files and export count table

``` r
## Create count tables from the individual raw count files generated by HTseq-count and load the metadata
# Create a vector holding all input files
inf <- dir("/HTseq_Count/Guts/", pattern="raw_counts.txt", full.names = TRUE)
inf
# Lapply will use read table on all input files, but will only report
# the second column (the counts)
count.table <- lapply(inf, function(f){
  read.table(f)[,2]
})
# do.call(cbind,...) binds list objects by columns to a matrix
count.table <- do.call(cbind,count.table)
# the rownames of the matrix are the entries of the first column
# of any file that htseq-count produces
rownames(count.table) <- read.table(inf[1])[,1]
# we remove the last 5 lines, as these are diagnostic messages
# from htseq-count, which we remove
count.table <- head(count.table, -5)

# Column names are the filenames minus all the prefix/suffix
colnames(count.table) <- gsub(basename(inf),
                              pattern="_raw_counts.txt",
                              replacement="")

#Remove genes with 0 total counts
count.table<-count.table[apply(count.table[,-1], 1, function(x) !all(x==0)),]

write.table(count.table, "~/Desktop/RNAseq_gut_counts.txt", sep="\t")
```

## Read in count table and metadata in R

``` r
count.table <- read.table("~/Desktop/RNAseq_gut_counts.txt", header = T)
count.table

Design <- read.table("~/Desktop/RNAseq_gut_metadata.txt", header = T, colClasses = "factor")
row.names(Design) <- Design$Sample_ID 
Design

# We make sure the order of the columns (Sample_IDs) in the count table match the metadata table
library(data.table)
count.table <- as.data.frame(count.table)
count.table <- setcolorder(count.table, as.character(Design$Sample_ID))
```

## DEG analysis with edgeR/Limma/Voom

``` r
library(edgeR)
y=DGEList(counts=count.table)

## Load annotation from gff file 
library(rtracklayer)
my_tags <- c("gene_id", "description", "transcript_id", "product")
df1 <- readGFF("/Users/joanitoliberti/Downloads/genome_assemblies_genome_gtf/ncbi-genomes-2020-12-28/GCF_003254395.2_Amel_HAv3.1_genomic.gtf", tags = my_tags)
head(df1)
df2 <- df1[,c("gene_id", "transcript_id")] 
df2 <- df2[!duplicated(df2), ]

delete.na <- function(DF, n=0) {
  DF[rowSums(is.na(DF)) <= n,]
}
df2<-delete.na(df2)

df3 <- df1[,c("gene_id", "product")] 
df3 <- transform(df3, n=nchar(as.character(product)))
df3 <- delete.na(df3)
df3 <- df3[with(df3, order(gene_id, n, decreasing = T)), ]
df3$n <- NULL
df3 <- df3[!duplicated(df3$gene_id), ]
df3$product <- sub(" isoform.*", "", df3$product)
df3$product <- sub(", transcript.*", "", df3$product)
df3$product <- sub(" precursor.*", "", df3$product)

annot<-df3[df3$gene_id%in%row.names(y),] 
annot <- annot[order(match(annot[,"gene_id"],row.names(y))),]

setdiff(row.names(y), annot$gene_id) # check that there are no differences between row names of y and merged annotation, otherwise it will not work to add the annotation to the DGElist 

# Append annotation to DGElist
gen <- as.data.frame(annot$product)
names(gen)[names(gen) == "annot$product"] <- "description"
y$genes <- gen

Hive=factor(Design$Hive)
Batch=factor(Design$Batch)
Bee_ID=factor(Design$Sample_ID)
Treatment <- relevel(Design$Treatment, "MD")
Treat=factor(Treatment)
```

## **Build the model**

``` r
#Set the model formula
design <- model.matrix(~0 + Treat + Batch)

#Filter low expressed genes
keep <- filterByExpr(y, design, min.count = 20, min.total.count = 20)
y <- y[keep,,keep.lib.sizes=FALSE]
dim(y)
```

    ## [1] 9679   38

``` r
#Now normalize counts
y=calcNormFactors(y, method = "TMM")

colnames(design)<-make.names(colnames(design))
v=voom(y,design,plot=TRUE)
```

![](Gut_RNAseqAnalysis_files/figure-gfm/unnamed-chunk-9-1.png)<!-- -->

``` r
fit=lmFit(v,design)
head(coef(fit))
```

    ##              TreatMD   TreatCL TreatCL_13 TreatCL_Bifi    Batch10    Batch11
    ## 18-w       3.7641449 5.2138719   5.375249     4.778698  0.6521125  0.6453782
    ## 5-HT1      4.8672318 4.6938249   4.632038     4.716229 -0.1009007  0.2885967
    ## 5-HT2alpha 2.6148944 2.8643069   2.657939     2.516012 -0.1452614  0.2821459
    ## 5-HT2beta  0.8724753 1.1849076   1.101570     1.073071  0.8275532  0.9924899
    ## 5-ht7      2.4745006 2.0422319   1.416110     1.869029 -0.7542617 -0.6718350
    ## A4         2.6430145 0.9158267   2.124185     1.543544  1.7385940  2.1828060
    ##                 Batch2      Batch3      Batch4      Batch5     Batch6
    ## 18-w        0.79154843  0.65677302 -0.31866969 -0.07935469  0.2575110
    ## 5-HT1      -0.03378571  0.50367745  0.42625089 -0.48108311 -0.2811201
    ## 5-HT2alpha  0.30765885  0.46700095 -0.06348462  0.02385823  0.1978182
    ## 5-HT2beta   0.66776135  0.79149333  0.79299976  0.40667065  0.7046536
    ## 5-ht7      -0.74858303 -0.09744168  1.00829290 -0.22439288 -0.9296316
    ## A4          1.42223804  0.65276638  0.87005560  0.18087056  1.2073310
    ##                 Batch7       Batch8       Batch9
    ## 18-w        1.12649201  0.415729201  1.191635981
    ## 5-HT1      -0.29611711  0.004202753  0.146765100
    ## 5-HT2alpha -0.04396406 -0.187056149  0.004484151
    ## 5-HT2beta   0.58778038  1.059909844  1.617851689
    ## 5-ht7      -0.84573449 -0.663701308 -0.810480659
    ## A4         -0.53607514 -0.147458280 -0.244943191

### **1) What genes are DEGs between the treatments overall? MD = microbiota-depleted, CL_Bifi = colonization with only Bifidobacterium, CL_13 = colonization with a community of 13 strains grown on plates, CL = colonization with a gut homogenate from 5 nurse bees**

``` r
con <- makeContrasts(
   MDvsCL=  TreatCL - TreatMD,
   MDvsCL_13=  TreatCL_13 - TreatMD,
   MDvsCLCL_13=  (TreatCL_13 + TreatCL)/2 - TreatMD,
   MDvsCL_Bifi=  TreatCL_Bifi - TreatMD,
   MDvsCLCL_13CL_Bifi=  (TreatCL_13 + TreatCL)/2 - TreatMD,
   CL_13vsCL=  TreatCL - TreatCL_13,
    levels=design)

fit2=contrasts.fit(fit,con)
fit2=eBayes(fit2, robust=TRUE)

dt.Treat <- decideTests(fit2, p=0.05)
summary(dt.Treat)
```

    ##        MDvsCL MDvsCL_13 MDvsCLCL_13 MDvsCL_Bifi MDvsCLCL_13CL_Bifi CL_13vsCL
    ## Down     1482      2240        2298         356               2298         0
    ## NotSig   6382      5093        5026        8319               5026      9679
    ## Up       1815      2346        2355        1004               2355         0

``` r
MDvsCL_table=topTable(fit2,coef="MDvsCL",n=Inf, sort="p", p=0.05)
MDvsCL_13_table=topTable(fit2,coef="MDvsCL_13",n=Inf, sort="p", p=0.05)
MDvsCLCL_13_table=topTable(fit2,coef="MDvsCLCL_13",n=Inf, sort="p", p=0.05)
MDvsCL_Bifi_table=topTable(fit2,coef="MDvsCL_Bifi",n=Inf, sort="p", p=0.05)
MDvsCLCL_13CL_Bifi_table=topTable(fit2,coef="MDvsCLCL_13CL_Bifi",n=Inf, sort="p", p=0.05)
CL_13vsCL_table=topTable(fit2,coef="CL_13vsCL",n=Inf, sort="p", p=0.05)

# Store the DEG list of each comparison in a vector 
MDvsCL_DEGs <- row.names(MDvsCL_table)
MDvsCL_13_DEGs <- row.names(MDvsCL_13_table)
MDvsCLCL_13_DEGs <- row.names(MDvsCLCL_13_table)
MDvsCL_Bifi_DEGs <- row.names(MDvsCL_Bifi_table)
MDvsCLCL_13CL_Bifi_DEGs <- row.names(MDvsCLCL_13CL_Bifi_table)
CL_13vsCL_DEGs <- row.names(CL_13vsCL_table)
AllDEGs <- unique(sort(c(MDvsCL_DEGs, MDvsCL_13_DEGs, MDvsCLCL_13_DEGs, MDvsCL_Bifi_DEGs, MDvsCLCL_13CL_Bifi_DEGs)))
```

### **MD vs. CL**

``` r
MDvsCL_table
```

### **MD vs. CL_13**

``` r
MDvsCL_13_table
```

### **MD vs. CL and CL_13**

``` r
MDvsCLCL_13_table
```

### **MD vs. CL_Bifi**

``` r
MDvsCL_Bifi_table
```

### **MD vs. CL, CL_13 and CL_Bifi**

``` r
MDvsCLCL_13CL_Bifi_table
```

### **CL_13 vs. CL**

``` r
CL_13vsCL_table
```

## Export tables to Excel file

``` r
library(xlsx)
write.xlsx(MDvsCL_table, file="~/Desktop/Gut_DEGresults.xlsx", sheetName="MDvsCL", row.names=TRUE)
write.xlsx(MDvsCL_13_table, file="~/Desktop/Gut_DEGresults.xlsx", sheetName="MDvsCL_13", append=TRUE, row.names=TRUE)
write.xlsx(MDvsCLCL_13_table, file="~/Desktop/Gut_DEGresults.xlsx", sheetName="MDvsCLCL_13", append=TRUE, row.names=TRUE)
write.xlsx(MDvsCL_Bifi_table, file="~/Desktop/Gut_DEGresults.xlsx", sheetName="MDvsCL_Bifi", append=TRUE, row.names=TRUE)
write.xlsx(MDvsCLCL_13CL_Bifi_table, file="~/Desktop/Gut_DEGresults.xlsx", sheetName="MDvsCLCL_13CL_Bifi", append=TRUE, row.names=TRUE)
```

## Remove batch effect (only for plotting)

``` r
lcpm <- cpm(y, log=TRUE, prior.count=3)
lcpm <- removeBatchEffect(lcpm, Design$Batch)
lcpmAllDEGs<-lcpm[row.names(lcpm)%in%AllDEGs,] 
```

## Plot gene expression profiles based on the list of DEGs in a principal component analysis

``` r
library(PCAtools)

p1 <- pca(lcpmAllDEGs, metadata = Design, removeVar = 0.1)

biplot(p1, lab = NULL, 
    colby = 'Treatment', 
    hline = 0, vline = 0,
    legendPosition = 'right') +
    scale_color_manual(values = c("#00BFC4","#7CAE00","#F8766D","#C77CFF"))
```

![](Gut_RNAseqAnalysis_files/figure-gfm/unnamed-chunk-19-1.png)<!-- -->

``` r
# ggsave(file="~/Desktop/All_gut_DEGs_Ordination.pdf", width=8, height=6, useDingbats=FALSE)
```

## **Venn diagram to compare overlap in DEG lists**

``` r
# Build a list containing vectors of DEGs in each comparison
DEGs_Treat <- list(MDvsCL_DEGs, MDvsCL_13_DEGs, MDvsCL_Bifi_DEGs)

# Rename list vectors
names(DEGs_Treat) <- c("MDvsCL", "MDvsCL_13", "MDvsCL_Bifi")

# Plot a Venn diagram with the VennDiagram R package
require("VennDiagram")
overrideTriple=T
venn.plot1 <- venn.diagram(DEGs_Treat, NULL, 
                           fill=c("darkmagenta", "green", "salmon"), 
                           alpha=c(0.5,0.5,0.5), 
                           cex = 1.6, 
                           cat.fontface=4, 
                           category.names=c("MDvsCL", "MDvsCL_13",  "MDvsCL_Bifi"), 
                           main="Differentially expressed genes")

# To plot the venn diagram we use the grid.draw() function 
grid.newpage()
# pdf("~/Desktop/Venn_Gut_gutmicrobiota3way_RNA-seq.pdf",height=4,width=6)
p1 <- grid.draw(venn.plot1)
```

![](Gut_RNAseqAnalysis_files/figure-gfm/unnamed-chunk-20-1.png)<!-- -->

``` r
# dev.off()
```

### Save the DEG list

``` r
write.table(AllDEGs, "~/Desktop/Gut_allDEGs.txt", sep="\t", col.names=NA)
```

## **Session info**

``` r
sessionInfo()
```

    ## R version 4.1.0 (2021-05-18)
    ## Platform: x86_64-apple-darwin17.0 (64-bit)
    ## Running under: macOS Mojave 10.14.6
    ## 
    ## Matrix products: default
    ## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
    ## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
    ## 
    ## locale:
    ## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
    ## 
    ## attached base packages:
    ##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
    ##  [8] datasets  methods   base     
    ## 
    ## other attached packages:
    ##  [1] VennDiagram_1.7.0    futile.logger_1.4.3  PCAtools_2.4.0      
    ##  [4] ggrepel_0.9.1        ggplot2_3.3.5        xlsx_0.6.5          
    ##  [7] rtracklayer_1.52.1   GenomicRanges_1.44.0 GenomeInfoDb_1.28.4 
    ## [10] IRanges_2.26.0       S4Vectors_0.30.2     BiocGenerics_0.38.0 
    ## [13] edgeR_3.34.1         limma_3.48.3         data.table_1.14.2   
    ## 
    ## loaded via a namespace (and not attached):
    ##  [1] bitops_1.0-7                matrixStats_0.61.0         
    ##  [3] tools_4.1.0                 utf8_1.2.2                 
    ##  [5] R6_2.5.1                    irlba_2.3.3                
    ##  [7] DBI_1.1.1                   colorspace_2.0-2           
    ##  [9] withr_2.4.2                 tidyselect_1.1.1           
    ## [11] compiler_4.1.0              Biobase_2.52.0             
    ## [13] formatR_1.11                DelayedArray_0.18.0        
    ## [15] labeling_0.4.2              scales_1.1.1               
    ## [17] stringr_1.4.0               digest_0.6.28              
    ## [19] Rsamtools_2.8.0             rmarkdown_2.11             
    ## [21] XVector_0.32.0              pkgconfig_2.0.3            
    ## [23] htmltools_0.5.2             sparseMatrixStats_1.4.2    
    ## [25] MatrixGenerics_1.4.3        fastmap_1.1.0              
    ## [27] highr_0.9                   rlang_0.4.12               
    ## [29] DelayedMatrixStats_1.14.3   BiocIO_1.2.0               
    ## [31] generics_0.1.1              farver_2.1.0               
    ## [33] BiocParallel_1.26.2         dplyr_1.0.7                
    ## [35] RCurl_1.98-1.5              magrittr_2.0.1             
    ## [37] BiocSingular_1.8.1          GenomeInfoDbData_1.2.6     
    ## [39] Matrix_1.3-4                Rcpp_1.0.7                 
    ## [41] munsell_0.5.0               fansi_0.5.0                
    ## [43] lifecycle_1.0.1             stringi_1.7.5              
    ## [45] yaml_2.2.1                  SummarizedExperiment_1.22.0
    ## [47] zlibbioc_1.38.0             plyr_1.8.6                 
    ## [49] dqrng_0.3.0                 crayon_1.4.2               
    ## [51] lattice_0.20-45             Biostrings_2.60.2          
    ## [53] cowplot_1.1.1               beachmat_2.8.1             
    ## [55] xlsxjars_0.6.1              locfit_1.5-9.4             
    ## [57] knitr_1.36                  pillar_1.6.4               
    ## [59] rjson_0.2.20                reshape2_1.4.4             
    ## [61] ScaledMatrix_1.0.0          futile.options_1.0.1       
    ## [63] XML_3.99-0.8                glue_1.5.0                 
    ## [65] evaluate_0.14               lambda.r_1.2.4             
    ## [67] vctrs_0.3.8                 gtable_0.3.0               
    ## [69] purrr_0.3.4                 assertthat_0.2.1           
    ## [71] xfun_0.28                   rsvd_1.0.5                 
    ## [73] restfulr_0.0.13             tibble_3.1.6               
    ## [75] rJava_1.0-5                 GenomicAlignments_1.28.0   
    ## [77] statmod_1.4.36              ellipsis_0.3.2
