Differential gene expression analysis of the brain of gnotobiotic honey
bees
================
Joanito Liberti, University of Lausanne

## Check data quality with FastQC

``` bash
for f in *.fastq.gz; do 
fastqc $f -o /work/FASTQC/; 
done
```

## Quality trimming with Trimmomatic

``` bash
# We use GNU parallel to run Trimmomatic on multiple fastq.gz files simultaneously
parallel 'java -jar /work/Software/Trimmomatic-0.39/trimmomatic-0.39.jar SE {} /work/07_Trimmomatic/{.}_trim.fastq.gz ILLUMINACLIP:/work/Software/Trimmomatic-0.39/adapters/TruSeq3-SE.fa:2:30:10 LEADING:10 TRAILING:10 SLIDINGWINDOW:4:20 MINLEN:80' ::: *.fastq.gz
```

## Map reads to the honey bee genome (Amel_HAv3.1) with STAR

``` bash
# Create genome indices, Amel_HAv3.1
STAR --runThreadN 10 --runMode genomeGenerate --genomeDir /work/02_genome/HAv3.1 --genomeFastaFiles /work/02_genome/HAv3.1/GCF_003254395.2_Amel_HAv3.1_genomic.fna --sjdbGTFfile /work/02_genome/HAv3.1/GCF_003254395.2_Amel_HAv3.1_genomic.gff --sjdbGTFfeatureExon exon --sjdbGTFtagExonParentTranscript Parent --sjdbOverhang 150 &

## Map Trimmomatic filtered fastq files against the Hav3.1 genome:
for f in *.fastq.gz
do
STAR --genomeDir /work/02_genome/HAv3.1/ --readFilesIn /work/07_Trimmomatic/Brains/$f --outFileNamePrefix /work/08_Alignments_HavGenome/Alignments_Brains_HAv3.1/$f --outSAMtype BAM SortedByCoordinate --limitBAMsortRAM 30000000000 --sjdbGTFfile /work/02_genome/HAv3.1/GCF_003254395.2_Amel_HAv3.1_genomic.gff --sjdbGTFfeatureExon exon --sjdbGTFtagExonParentTranscript transcript_id --readFilesCommand zcat --sjdbOverhang 150 --runThreadN 20
done
```

## As each sample was sequenced across two lanes we need to merge the .bam files

``` bash
printf '%s\0' *.bam | sort -zV | xargs -0 -n2 sh -c 'out="${1%_L*}"; echo samtools merge -@ 20 "${out}.bam" "$1" "$2"' sh # first check that the command would be fine, then remove echo. -@ 20 means use 20 threads
```

## Count features with HTseq-count

``` bash
time (for f in *.bam
do 
echo ${f}
htseq-count -f bam -r pos -s no -m union -t exon -i gene_id $f /work/02_genome/HAv3.1/GCF_003254395.2_Amel_HAv3.1_genomic.gtf > ${f/.bam/_raw_counts.txt}
done)

# Remove first lines as these are the total counts
for f in *txt
do
tail -n +2 $f > $f.tmp && mv $f.tmp $f
done
```

## Merge count files and export count table

``` r
## Create count tables from the individual raw count files generated by HTseq-count and load the metadata
# Create a vector holding all input files
inf <- dir("/HTseq_Count/Brains/", pattern="raw_counts.txt", full.names = TRUE)
inf
# Lapply will use read table on all input files, but will only report
# the second column (the counts)
count.table <- lapply(inf, function(f){
  read.table(f)[,2]
})
# do.call(cbind,...) binds list objects by columns to a matrix
count.table <- do.call(cbind,count.table)
# the rownames of the matrix are the entries of the first column
# of any file that htseq-count produces
rownames(count.table) <- read.table(inf[1])[,1]
# we remove the last 5 lines, as these are diagnostic messages
# from htseq-count, which we remove
count.table <- head(count.table, -5)

# Column names are the filenames minus all the prefix/suffix
colnames(count.table) <- gsub(basename(inf),
                              pattern="_raw_counts.txt",
                              replacement="")

#Remove genes with 0 total counts
count.table<-count.table[apply(count.table[,-1], 1, function(x) !all(x==0)),]

write.table(count.table, "~/Desktop/RNAseq_brain_counts.txt", sep="\t")
```

# Read in count table and metadata in R

``` r
count.table <- read.table("~/Desktop/RNAseq_brain_counts.txt", header = T, check.names = F)
count.table

Design <- read.table("~/Desktop/RNAseq_brain_metadata.txt", header = T, colClasses = "factor")
Design

Design_AL <- Design[Design$Region == "AL", ]
Design_MB <- Design[Design$Region == "MB", ]
Design_OL <- Design[Design$Region == "OL", ]

# We make sure the order of the columns (Sample_IDs) in the count table matches that of the metadata table
library(data.table)
count.table <- as.data.frame(count.table)
count.table <- setcolorder(count.table, as.character(Design$Sample))
```

## DEG analysis with edgeR/Limma/Voom

``` r
library(edgeR)
y=DGEList(counts=count.table)

## Load gene annotation from gff file 
library(rtracklayer)
my_tags <- c("gene_id", "description", "transcript_id", "product")
df1 <- readGFF("/Users/joanitoliberti/Downloads/genome_assemblies_genome_gtf/ncbi-genomes-2020-12-28/GCF_003254395.2_Amel_HAv3.1_genomic.gtf", tags = my_tags)
head(df1)
df2 <- df1[,c("gene_id", "transcript_id")] 
df2 <- df2[!duplicated(df2), ]

delete.na <- function(DF, n=0) {
  DF[rowSums(is.na(DF)) <= n,]
}
df2<-delete.na(df2)

df3 <- df1[,c("gene_id", "product")] 
df3 <- transform(df3, n=nchar(as.character(product)))
df3 <- delete.na(df3)
df3 <- df3[with(df3, order(gene_id, n, decreasing = T)), ]
df3$n <- NULL
df3 <- df3[!duplicated(df3$gene_id), ]
df3$product <- sub(" isoform.*", "", df3$product)
df3$product <- sub(", transcript.*", "", df3$product)
df3$product <- sub(" precursor.*", "", df3$product)

annot<-df3[df3$gene_id%in%row.names(y),] 
annot <- annot[order(match(annot[,"gene_id"],row.names(y))),]

setdiff(row.names(y), annot$gene_id) # check that there are no differences between row names of y and merged annotation, otherwise it will not work to add the annotation to the DGElist 

# Append annotation to DGElist
gen <- as.data.frame(annot$product)
names(gen)[names(gen) == "annot$product"] <- "description"
y$genes <- gen

Hive=factor(Design$Hive)
Batch=factor(Design$Batch)
Bee_ID=factor(Design$Sample_ID)
Region=factor(Design$Region)
Treatment <- relevel(Design$Treatment, "MD")
Treat=factor(Treatment)
```

## **Build the model**

``` r
# Set the model formula - Here we test all samples together and then we will take into account that the regions are paired by bee individual. We make one group factor containing all combinations of treatments and brain regions
group <- paste0(Treat, Region) 
design <- model.matrix(~0 + group + Batch)

# Filter out low expressed genes
keep <- filterByExpr(y, design, min.count = 20, min.total.count = 20)
y <- y[keep,,keep.lib.sizes=FALSE]
dim(y)
```

    ## [1] 10493   116

``` r
# Now normalize counts
y=calcNormFactors(y, method = "TMM")

colnames(design)<-make.names(colnames(design))
v=voom(y,design,plot=TRUE)
```

![](02-Brain-RNAseqAnalysis_files/figure-gfm/unnamed-chunk-9-1.png)<!-- -->

``` r
cor=duplicateCorrelation(v,design,block=Design$Sample_ID)

fit=lmFit(v,design,block=Design$Sample_ID, correlation=cor$consensus)
head(coef(fit))
```

    ##            groupCL_13AL groupCL_13MB groupCL_13OL groupCL_BifiAL groupCL_BifiMB
    ## 18-w           6.656287     7.138677     6.535407       6.540522       7.290839
    ## 5-HT1          6.961588     8.823847     7.304363       6.927082       8.896708
    ## 5-HT2alpha     3.925708     5.826524     4.254083       3.962462       5.892414
    ## 5-HT2beta      3.065348     1.516185     2.522427       2.844925       1.588782
    ## 5-ht7          5.101798     3.708694     4.213428       5.071945       3.697706
    ## A4             2.736842     3.101386     2.573705       2.700543       2.385979
    ##            groupCL_BifiOL groupCLAL groupCLMB groupCLOL groupMDAL groupMDMB
    ## 18-w             6.498666  6.686847  7.280763  6.575698  6.583331  7.234389
    ## 5-HT1            7.245293  7.094621  8.958955  7.337585  6.981090  8.842046
    ## 5-HT2alpha       4.097792  4.008248  6.045440  4.182330  4.000623  5.944695
    ## 5-HT2beta        2.496309  3.110320  1.894322  2.529677  2.968311  1.774069
    ## 5-ht7            4.251922  5.025366  3.723848  4.299298  5.038624  3.852961
    ## A4               2.182551  2.804679  2.787749  2.578917  2.692906  2.922841
    ##            groupMDOL     Batch10     Batch11      Batch2      Batch3
    ## 18-w        6.466668  0.15698147  0.38798576  0.04687339 -0.05721032
    ## 5-HT1       7.372537  0.09650947  0.40646323  0.12694402 -0.08229683
    ## 5-HT2alpha  4.317770 -0.06700991  0.29468863 -0.18534204 -0.10597280
    ## 5-HT2beta   2.511350  0.15008552  0.21955994  0.12307522 -0.18357515
    ## 5-ht7       4.166463 -0.15518956  0.05358565 -0.09981766 -0.21344633
    ## A4          2.045632 -0.43345741 -0.20554588 -0.53761045  0.41596494
    ##                 Batch4      Batch5       Batch6       Batch7      Batch8
    ## 18-w       -0.14416961  0.08298738 -0.330363372 -0.562710622  0.11123063
    ## 5-HT1      -0.06056591 -0.01764405 -0.003445501 -0.001554369  0.17479422
    ## 5-HT2alpha -0.05726305 -0.33406793 -0.213445528 -0.759418760  0.13493842
    ## 5-HT2beta   0.02148560  0.11162414 -0.562017408 -1.006586716  0.04735407
    ## 5-ht7      -0.06469804 -0.17700820  0.089007788 -0.196424446 -0.11178786
    ## A4          0.49728844 -0.16530366 -0.769839585 -0.728625922 -0.35199470
    ##                 Batch9
    ## 18-w       -0.06662617
    ## 5-HT1      -0.06229095
    ## 5-HT2alpha -0.26115253
    ## 5-HT2beta   0.15358421
    ## 5-ht7      -0.08493610
    ## A4         -1.12230099

### Remove batch effect (only for plotting)

``` r
library("grid")
library("ggplotify")

lcpm <- cpm(y, log=TRUE, prior.count=3)
lcpm <- removeBatchEffect(lcpm, Design$Batch)

lcpm_AL<-lcpm[,colnames(lcpm)%in%Design_AL[,1]] 
lcpm_MB<-lcpm[,colnames(lcpm)%in%Design_MB[,1]] 
lcpm_OL<-lcpm[,colnames(lcpm)%in%Design_OL[,1]] 
```

# Plot ordination

``` r
library(PCAtools)

row.names(Design) <- Design[,1]
p <- pca(lcpm, metadata = Design, removeVar = 0.1)

biplot(p, lab = NULL, 
    colby = 'Treatment', 
    shape = 'Region',
    hline = 0, vline = 0,
    legendPosition = 'right') +
    scale_color_manual(values = c("#00BFC4","#7CAE00","#F8766D","#C77CFF"))
```

![](02-Brain-RNAseqAnalysis_files/figure-gfm/unnamed-chunk-11-1.png)<!-- -->

``` r
# ggsave(file="~/Desktop/OverallBrainGeneExpression.pdf", width=8, height=6, useDingbats=FALSE)
```

## **1) What genes are DEGs between the treatments overall (across the three brain regions)? MD = microbiota-depleted bees, CL_Bifi = bees colonized only with Bifidobacterium, CL_13 = bees colonized with a community of 13 strains grown on plates, CL = bees colonized with a gut homogenate from 5 nurse bees**

``` r
con <- makeContrasts(
   MDvsCL=  (groupCLAL + groupCLMB + groupCLOL)/3 -
    (groupMDAL + groupMDMB + groupMDOL)/3,
   MDvsCL_13=  (groupCL_13AL + groupCL_13MB + groupCL_13OL)/3 -
    (groupMDAL + groupMDMB + groupMDOL)/3,
   MDvsCLCL_13=  (groupCL_13AL + groupCL_13MB + groupCL_13OL +
                 groupCLAL + groupCLMB + groupCLOL)/6 -
                (groupMDAL + groupMDMB + groupMDOL)/3,
   MDvsCL_Bifi=  (groupCL_BifiAL + groupCL_BifiMB + groupCL_BifiOL)/3 -
    (groupMDAL + groupMDMB + groupMDOL)/3,
   MDvsCLCL_13CL_Bifi=  (groupCL_13AL + groupCL_13MB + groupCL_13OL +
                 groupCLAL + groupCLMB + groupCLOL +
                   groupCL_BifiAL + groupCL_BifiMB + groupCL_BifiOL)/9 -
                (groupMDAL + groupMDMB + groupMDOL)/3,
   CL_13vsCL=  (groupCLAL + groupCLMB + groupCLOL)/3 -
    (groupCL_13AL + groupCL_13MB + groupCL_13OL)/3,
    levels=design)

fit2=contrasts.fit(fit,con)
fit2=eBayes(fit2, robust=TRUE)

dt.Treat <- decideTests(fit2, p=0.05)
summary(dt.Treat)

MDvsCL_table=topTable(fit2,coef="MDvsCL",n=Inf, sort="p", p=0.05)
MDvsCL_13_table=topTable(fit2,coef="MDvsCL_13",n=Inf, sort="p", p=0.05)
MDvsCLCL_13_table=topTable(fit2,coef="MDvsCLCL_13",n=Inf, sort="p", p=0.05)
MDvsCL_Bifi_table=topTable(fit2,coef="MDvsCL_Bifi",n=Inf, sort="p", p=0.05)
MDvsCLCL_13CL_Bifi_table=topTable(fit2,coef="MDvsCLCL_13CL_Bifi",n=Inf, sort="p", p=0.05)
CL_13vsCL_table=topTable(fit2,coef="CL_13vsCL",n=Inf, sort="p", p=0.05)

# Store the DEG lists of each contrast 
MDvsCL_DEGs <- row.names(MDvsCL_table)
MDvsCL_13_DEGs <- row.names(MDvsCL_13_table)
MDvsCLCL_13_DEGs <- row.names(MDvsCLCL_13_table)
MDvsCL_Bifi_DEGs <- row.names(MDvsCL_Bifi_table)
MDvsCLCL_13CL_Bifi_DEGs <- row.names(MDvsCLCL_13CL_Bifi_table)
CL_13vsCL_DEGs <- row.names(CL_13vsCL_table)

AcrossBrain_DEGs <- unique(sort(c(MDvsCL_DEGs, MDvsCL_13_DEGs, MDvsCLCL_13_DEGs, MDvsCL_Bifi_DEGs, MDvsCLCL_13CL_Bifi_DEGs)))
```

### **MD vs. CL**

``` r
MDvsCL_table
```

    ##                                                                                                             description
    ## LOC551518                                                                                       glutamyl aminopeptidase
    ## LOC410509                                            uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC102654070                                                                               uncharacterized LOC102654070
    ## LOC412034                                                                                             DNA helicase MCM8
    ## Mrjp3                                                                                       major royal jelly protein 3
    ## LOC113218542                                                                               uncharacterized LOC113218542
    ## LOC102656292                                                                               uncharacterized LOC102656292
    ## LOC413924                                                                      glyceraldehyde-3-phosphate dehydrogenase
    ## Mrjp1                                                                                       major royal jelly protein 1
    ## Calpb                                                                                                         calpain-B
    ## LOC726642                                                                                      DNA repair protein RAD50
    ## LOC412863                                                                                       histone demethylase UTY
    ## LOC551109                                                                                   kinesin-like protein Klp61F
    ## LOC409222                                                                                                      cyclin G
    ## LOC552557    lipoamide acyltransferase component of branched-chain alpha-keto acid dehydrogenase complex, mitochondrial
    ## LOC100578353                                                                               uncharacterized LOC100578353
    ## LOC100577678                                                              putative uncharacterized protein DDB_G0286901
    ## LOC100576157                                                                       uncharacterized protein LOC100576157
    ## LOC552342                                                                            lysophospholipid acyltransferase 2
    ## LOC411917                                                                   NAD-dependent protein deacetylase sirtuin-1
    ## LOC113219349                                                                               uncharacterized LOC113219349
    ## LOC551869                                                                         vitamin K-dependent gamma-carboxylase
    ## LOC410550                                                                         aminomethyltransferase, mitochondrial
    ## LOC113219061                                                                               uncharacterized LOC113219061
    ## LOC409572                                                                   TWiK family of potassium channels protein 9
    ## LOC100577833                                                                                               histone H1.2
    ## LOC102656664                                                                               uncharacterized LOC102656664
    ## LOC412832                                                               structural maintenance of chromosomes protein 2
    ## LOC107965692                                                                            poly(ADP-ribose) glycohydrolase
    ## LOC551849                                                                                     L-galactose dehydrogenase
    ##                   logFC    AveExpr         t      P.Value    adj.P.Val
    ## LOC551518     0.2587229  6.1667524  6.294046 7.921517e-09 5.541397e-05
    ## LOC410509    -2.0081512 -0.6609070 -6.267020 1.056208e-08 5.541397e-05
    ## LOC102654070 -1.7251557 -0.3473674 -6.121501 1.755172e-08 6.139007e-05
    ## LOC412034    -1.7006591  3.0343159 -5.522324 2.908472e-07 6.607609e-04
    ## Mrjp3         4.1738533  0.4812802  5.507008 3.148579e-07 6.607609e-04
    ## LOC113218542  1.4079168  2.7427661  5.272598 8.423331e-07 1.473100e-03
    ## LOC102656292 -0.5600821  3.0676037 -4.930488 3.201673e-06 4.799307e-03
    ## LOC413924    -1.1185329  1.9163459 -4.786306 5.973768e-06 7.835343e-03
    ## Mrjp1         4.3471949  3.2045800  4.705297 8.680430e-06 1.012042e-02
    ## Calpb        -0.1793846  8.0502586 -4.476117 1.989605e-05 2.087693e-02
    ## LOC726642    -0.2774192  6.4228412 -4.452129 2.204113e-05 2.102523e-02
    ## LOC412863    -0.4661243  5.9201781 -4.435476 2.448719e-05 2.141200e-02
    ## LOC551109    -0.3014156  4.5016549 -4.397800 2.697709e-05 2.177466e-02
    ## LOC409222     0.1459867  7.1555059  4.345405 3.301234e-05 2.297422e-02
    ## LOC552557    -0.2387171  4.3380585 -4.340676 3.361695e-05 2.297422e-02
    ## LOC100578353 -0.9295838  0.6911799 -4.312623 3.742953e-05 2.297422e-02
    ## LOC100577678  0.8627111  1.3591698  4.287478 4.132544e-05 2.297422e-02
    ## LOC100576157  0.4617116  3.3021060  4.283116 4.189088e-05 2.297422e-02
    ## LOC552342    -0.2534476  5.6924921 -4.270016 4.402732e-05 2.297422e-02
    ## LOC411917     0.1884807  6.7528029  4.262516 4.529913e-05 2.297422e-02
    ## LOC113219349 -2.0969843 -0.7986247 -4.266380 4.597908e-05 2.297422e-02
    ## LOC551869     0.1829174  6.9319825  4.229403 5.134907e-05 2.449117e-02
    ## LOC410550     0.4430775  3.8910897  4.161687 6.719095e-05 2.952601e-02
    ## LOC113219061 -0.5666056  2.4034193 -4.156476 6.753304e-05 2.952601e-02
    ## LOC409572     0.8375308  0.5828705  4.096957 8.426979e-05 3.536972e-02
    ## LOC100577833 -0.9626887  5.4159446 -4.075872 9.558800e-05 3.857711e-02
    ## LOC102656664 -0.5827152  1.2696639 -4.013506 1.145586e-04 4.379386e-02
    ## LOC412832    -0.2692453  4.8122791 -4.008060 1.168615e-04 4.379386e-02
    ## LOC107965692 -0.1550595  6.2832338 -3.983212 1.279412e-04 4.629266e-02
    ## LOC551849    -0.3188634  2.9979637 -3.966319 1.360403e-04 4.758237e-02
    ##                       B
    ## LOC551518     9.9350661
    ## LOC410509     8.3586698
    ## LOC102654070  4.5352380
    ## LOC412034     6.1441174
    ## Mrjp3         5.9897369
    ## LOC113218542  5.3791722
    ## LOC102656292  4.1609353
    ## LOC413924     3.1731878
    ## Mrjp1         3.3961888
    ## Calpb         2.4009029
    ## LOC726642     2.3469870
    ## LOC412863     2.2818653
    ## LOC551109     2.3372834
    ## LOC409222     1.9372225
    ## LOC552557     2.1241094
    ## LOC100578353  1.0723145
    ## LOC100577678  1.4530542
    ## LOC100576157  1.9579343
    ## LOC552342     1.7423415
    ## LOC411917     1.6532537
    ## LOC113219349 -0.1172664
    ## LOC551869     1.5378666
    ## LOC410550     1.5129954
    ## LOC113219061  1.4353988
    ## LOC409572     0.4533019
    ## LOC100577833  1.1024192
    ## LOC102656664  0.5733556
    ## LOC412832     0.9595956
    ## LOC107965692  0.6984447
    ## LOC551849     0.8884507

### **MD vs. CL_13**

``` r
MDvsCL_13_table
```

    ##                                                                              description
    ## LOC410509             uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC412034                                                              DNA helicase MCM8
    ## LOC551518                                                        glutamyl aminopeptidase
    ## LOC102654070                                                uncharacterized LOC102654070
    ## LOC100577359                                                uncharacterized LOC100577359
    ## LOC113218542                                                uncharacterized LOC113218542
    ## LOC100578804                                                uncharacterized LOC100578804
    ## LOC107965822 retrovirus-related Pol polyprotein from type-1 retrotransposable element R2
    ## LOC100576976                putative leucine-rich repeat-containing protein DDB_G0290503
    ## LOC552037                                              POC1 centriolar protein homolog A
    ## LOC113219194                                                uncharacterized LOC113219194
    ## LOC412863                                                        histone demethylase UTY
    ## LOC100576732                                                uncharacterized LOC100576732
    ## LOC107965375                                                uncharacterized LOC107965375
    ## LOC102654286                                                           trichohyalin-like
    ## LOC107965692                                             poly(ADP-ribose) glycohydrolase
    ## LOC102656557                                                uncharacterized LOC102656557
    ## LOC100576828                                                   protein maelstrom homolog
    ## LOC412528                                                          eyes absent homolog 4
    ## LOC551847                                                  N-acetyl-D-glucosamine kinase
    ## LOC102654369                                                        la-related protein 6
    ## LOC727293                                                              guanine deaminase
    ## LOC409023                                         methylthioribose-1-phosphate isomerase
    ## LOC100576447                     DNA damage-regulated autophagy modulator protein 2-like
    ##                   logFC    AveExpr         t      P.Value    adj.P.Val
    ## LOC410509    -3.3371069 -0.6609070 -9.023469 1.929443e-14 2.024564e-10
    ## LOC412034    -2.0193466  3.0343159 -6.696193 1.497939e-09 7.858938e-06
    ## LOC551518     0.2253430  6.1667524  5.507833 2.746598e-07 9.606685e-04
    ## LOC102654070 -1.4600728 -0.3473674 -5.323203 6.115471e-07 1.604241e-03
    ## LOC100577359  1.7219310  1.0544558  5.279573 7.803021e-07 1.637542e-03
    ## LOC113218542  1.3522286  2.7427661  5.161215 1.343570e-06 2.124921e-03
    ## LOC100578804 -0.9400053  2.5333837 -5.133142 1.417559e-06 2.124921e-03
    ## LOC107965822 -0.8687329  5.1336058 -5.037284 2.261542e-06 2.966295e-03
    ## LOC100576976 -0.4557146  3.2910845 -4.949591 2.958597e-06 3.449395e-03
    ## LOC552037    -0.3612769  4.1189553 -4.857837 4.316420e-06 4.529220e-03
    ## LOC113219194  0.7176461  1.5508570  4.660587 9.592607e-06 9.150475e-03
    ## LOC412863    -0.4734449  5.9201781 -4.545787 1.596649e-05 1.396137e-02
    ## LOC100576732 -0.6377215  2.4041118 -4.511361 1.733055e-05 1.398842e-02
    ## LOC107965375 -0.7273658  2.1393904 -4.380052 2.900957e-05 2.174267e-02
    ## LOC102654286 -0.8599412  0.8498086 -4.323526 3.590067e-05 2.488915e-02
    ## LOC107965692 -0.1661187  6.2832338 -4.293176 4.031329e-05 2.488915e-02
    ## LOC102656557 -0.5855406  1.1036187 -4.291309 4.060113e-05 2.488915e-02
    ## LOC100576828 -0.5772911  2.7425116 -4.278096 4.269558e-05 2.488915e-02
    ## LOC412528     0.2135022  5.7952845  4.246350 4.816174e-05 2.592749e-02
    ## LOC551847    -0.4464815  3.6071100 -4.241041 4.941864e-05 2.592749e-02
    ## LOC102654369 -0.2632782  6.1465990 -4.216218 5.424888e-05 2.681324e-02
    ## LOC727293    -0.5516994  3.4666704 -4.208853 5.621760e-05 2.681324e-02
    ## LOC409023     0.2406309  5.5069228  4.173341 6.340350e-05 2.892578e-02
    ## LOC100576447 -0.7163391  1.9663377 -4.028435 1.096312e-04 4.793167e-02
    ##                       B
    ## LOC410509    17.2064817
    ## LOC412034    10.6418349
    ## LOC551518     6.5463708
    ## LOC102654070  2.5306597
    ## LOC100577359  3.7777764
    ## LOC113218542  4.9548957
    ## LOC100578804  4.6244549
    ## LOC107965822  4.5797673
    ## LOC100576976  4.3152570
    ## LOC552037     4.0322353
    ## LOC113219194  2.4158645
    ## LOC412863     2.6986502
    ## LOC100576732  2.5358167
    ## LOC107965375  1.9999024
    ## LOC102654286  1.0594533
    ## LOC107965692  1.7997025
    ## LOC102656557  1.3684306
    ## LOC100576828  1.8136280
    ## LOC412528     1.6769269
    ## LOC551847     1.8149787
    ## LOC102654369  1.5375017
    ## LOC727293     1.6914413
    ## LOC409023     1.4488313
    ## LOC100576447  0.9660734

### **MD vs. CL and CL_13**

``` r
MDvsCLCL_13_table
```

    ##                                                                     description
    ## LOC410509    uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC412034                                                     DNA helicase MCM8
    ## LOC551518                                               glutamyl aminopeptidase
    ## LOC102654070                                       uncharacterized LOC102654070
    ## LOC113218542                                       uncharacterized LOC113218542
    ## LOC412863                                               histone demethylase UTY
    ## LOC413924                              glyceraldehyde-3-phosphate dehydrogenase
    ## LOC107965692                                    poly(ADP-ribose) glycohydrolase
    ## LOC100578353                                       uncharacterized LOC100578353
    ## LOC409222                                                              cyclin G
    ## LOC113219061                                       uncharacterized LOC113219061
    ## LOC107965375                                       uncharacterized LOC107965375
    ## LOC551847                                         N-acetyl-D-glucosamine kinase
    ## LOC102656292                                       uncharacterized LOC102656292
    ## LOC552342                                    lysophospholipid acyltransferase 2
    ## LOC100577359                                       uncharacterized LOC100577359
    ## LOC725710                   D-beta-hydroxybutyrate dehydrogenase, mitochondrial
    ## LOC551936                               facilitated trehalose transporter Tret1
    ## LOC551109                                           kinesin-like protein Klp61F
    ## LOC100578804                                       uncharacterized LOC100578804
    ## LOC412528                                                 eyes absent homolog 4
    ## LOC552683                                 chitinase domain-containing protein 1
    ## LOC727293                                                     guanine deaminase
    ## Calpb                                                                 calpain-B
    ## LOC552069          probable hydroxyacid-oxoacid transhydrogenase, mitochondrial
    ## LOC410626                          sodium-coupled monocarboxylate transporter 1
    ## LOC726729                                           helix-loop-helix protein 11
    ##                   logFC    AveExpr         t      P.Value    adj.P.Val
    ## LOC410509    -2.6726291 -0.6609070 -9.120084 1.198203e-14 1.257274e-10
    ## LOC412034    -1.8600029  3.0343159 -7.119158 2.063018e-10 1.082363e-06
    ## LOC551518     0.2420330  6.1667524  6.724180 1.051117e-09 2.911717e-06
    ## LOC102654070 -1.5926143 -0.3473674 -6.712712 1.109966e-09 2.911717e-06
    ## LOC113218542  1.3800727  2.7427661  5.728190 1.188708e-07 2.494623e-04
    ## LOC412863    -0.4697846  5.9201781 -5.143887 1.423409e-06 2.489305e-03
    ## LOC413924    -1.0040934  1.9163459 -5.084093 1.758965e-06 2.636689e-03
    ## LOC107965692 -0.1605891  6.2832338 -4.731350 7.218580e-06 8.778783e-03
    ## LOC100578353 -0.8590282  0.6911799 -4.720887 7.529691e-06 8.778783e-03
    ## LOC409222     0.1351653  7.1555059  4.603775 1.203104e-05 1.123878e-02
    ## LOC113219061 -0.5387771  2.4034193 -4.593266 1.254359e-05 1.123878e-02
    ## LOC107965375 -0.6597484  2.1393904 -4.588331 1.285289e-05 1.123878e-02
    ## LOC551847    -0.4146323  3.6071100 -4.555905 1.464816e-05 1.182332e-02
    ## LOC102656292 -0.4282569  3.0676037 -4.429053 2.389981e-05 1.791291e-02
    ## LOC552342    -0.2238907  5.6924921 -4.328325 3.524697e-05 2.358231e-02
    ## LOC100577359  1.2850493  1.0544558  4.318396 3.768637e-05 2.358231e-02
    ## LOC725710    -0.7504391  2.8369133 -4.317305 3.825061e-05 2.358231e-02
    ## LOC551936    -0.2580506  3.7671536 -4.292263 4.045378e-05 2.358231e-02
    ## LOC551109    -0.2486367  4.5016549 -4.191593 5.920797e-05 3.269838e-02
    ## LOC100578804 -0.6446887  2.5333837 -4.147003 7.103945e-05 3.551964e-02
    ## LOC412528     0.1833001  5.7952845  4.142737 7.108667e-05 3.551964e-02
    ## LOC552683    -0.1900984  4.6499202 -4.101883 8.274605e-05 3.844671e-02
    ## LOC727293    -0.4643985  3.4666704 -4.100179 8.427277e-05 3.844671e-02
    ## Calpb        -0.1427827  8.0502586 -4.082364 8.894381e-05 3.888697e-02
    ## LOC552069    -0.4760867  1.3017701 -4.033012 1.066625e-04 4.476839e-02
    ## LOC410626     0.8841262  3.6428424  4.022704 1.155013e-04 4.538637e-02
    ## LOC726729    -0.2235526  5.5423792 -4.008658 1.167857e-04 4.538637e-02
    ##                       B
    ## LOC410509    17.9800338
    ## LOC412034    12.1751349
    ## LOC551518    11.8834355
    ## LOC102654070  5.6404263
    ## LOC113218542  6.9252586
    ## LOC412863     5.0147018
    ## LOC413924     4.0278485
    ## LOC107965692  3.4554557
    ## LOC100578353  2.0218903
    ## LOC409222     2.9451687
    ## LOC113219061  2.7858432
    ## LOC107965375  2.5953649
    ## LOC551847     2.9066366
    ## LOC102656292  2.3668175
    ## LOC552342     1.9977068
    ## LOC100577359  0.8186483
    ## LOC725710     1.9654415
    ## LOC551936     1.9852214
    ## LOC551109     1.6346082
    ## LOC100578804  1.3260777
    ## LOC412528     1.3476830
    ## LOC552683     1.3039253
    ## LOC727293     1.3260886
    ## Calpb         1.0445585
    ## LOC552069     0.3123618
    ## LOC410626     1.0457456
    ## LOC726729     0.8979966

### **MD vs. CL_Bifi**

``` r
MDvsCL_Bifi_table
```

    ##                                                                              description
    ## LOC725217                                               uncharacterized protein PFB0145c
    ## LOC102654070                                                uncharacterized LOC102654070
    ## Mrjp3                                                        major royal jelly protein 3
    ## LOC412034                                                              DNA helicase MCM8
    ## LOC551518                                                        glutamyl aminopeptidase
    ## LOC551908                                      cilia- and flagella-associated protein 47
    ## LOC102655710                                                  basic proline-rich protein
    ## LOC107965822 retrovirus-related Pol polyprotein from type-1 retrotransposable element R2
    ## LOC102656292                                                uncharacterized LOC102656292
    ## LOC413924                                       glyceraldehyde-3-phosphate dehydrogenase
    ## LOC107965859                                                uncharacterized LOC107965859
    ## LOC102654106                                                uncharacterized LOC102654106
    ## LOC102654507                                                uncharacterized LOC102654507
    ##                   logFC    AveExpr         t      P.Value    adj.P.Val
    ## LOC725217    -2.4245826  1.9953126 -6.838419 7.585897e-10 7.959882e-06
    ## LOC102654070 -1.6273030 -0.3473674 -5.817260 6.975891e-08 3.659901e-04
    ## Mrjp3         4.1351145  0.4812802  5.514966 3.042793e-07 8.465199e-04
    ## LOC412034    -1.6949451  3.0343159 -5.498168 3.226989e-07 8.465199e-04
    ## LOC551518     0.2175279  6.1667524  5.261254 7.975129e-07 1.673661e-03
    ## LOC551908     0.6811373  4.2702270  4.841224 4.919258e-06 8.261310e-03
    ## LOC102655710  1.2305317  3.8058474  4.818313 5.511214e-06 8.261310e-03
    ## LOC107965822 -0.8190762  5.1336058 -4.725320 7.992869e-06 1.048365e-02
    ## LOC102656292 -0.5285235  3.0676037 -4.622390 1.117248e-05 1.302587e-02
    ## LOC413924    -1.0624777  1.9163459 -4.582926 1.344925e-05 1.411229e-02
    ## LOC107965859  0.5143245  2.8233734  4.511725 1.730583e-05 1.650819e-02
    ## LOC102654106 -3.1843383 -3.0515065 -4.286389 4.225226e-05 3.691306e-02
    ## LOC102654507 -0.5052179  3.2555583 -4.260007 4.573237e-05 3.691306e-02
    ##                      B
    ## LOC725217    10.901197
    ## LOC102654070  4.864620
    ## Mrjp3         6.294154
    ## LOC412034     6.281234
    ## LOC551518     5.426652
    ## LOC551908     3.873197
    ## LOC102655710  3.800185
    ## LOC107965822  3.273436
    ## LOC102656292  3.145881
    ## LOC413924     2.778576
    ## LOC107965859  2.657824
    ## LOC102654106 -1.339297
    ## LOC102654507  1.883029

### **MD vs. CL, CL_13 and CL_Bifi**

``` r
MDvsCLCL_13CL_Bifi_table
```

    ##                                                                              description
    ## LOC410509             uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC412034                                                              DNA helicase MCM8
    ## LOC102654070                                                uncharacterized LOC102654070
    ## LOC551518                                                        glutamyl aminopeptidase
    ## LOC413924                                       glyceraldehyde-3-phosphate dehydrogenase
    ## LOC102656292                                                uncharacterized LOC102656292
    ## LOC100578353                                                uncharacterized LOC100578353
    ## LOC107965375                                                uncharacterized LOC107965375
    ## LOC107965822 retrovirus-related Pol polyprotein from type-1 retrotransposable element R2
    ## LOC412863                                                        histone demethylase UTY
    ## Mrjp3                                                        major royal jelly protein 3
    ## LOC113219061                                                uncharacterized LOC113219061
    ## LOC107965692                                             poly(ADP-ribose) glycohydrolase
    ## LOC113218542                                                uncharacterized LOC113218542
    ## LOC409222                                                                       cyclin G
    ## LOC725710                            D-beta-hydroxybutyrate dehydrogenase, mitochondrial
    ## LOC551936                                        facilitated trehalose transporter Tret1
    ## LOC551847                                                  N-acetyl-D-glucosamine kinase
    ## LOC552069                   probable hydroxyacid-oxoacid transhydrogenase, mitochondrial
    ## LOC102656664                                                uncharacterized LOC102656664
    ## LOC727293                                                              guanine deaminase
    ## Calpb                                                                          calpain-B
    ## LOC100578804                                                uncharacterized LOC100578804
    ##                   logFC    AveExpr         t      P.Value    adj.P.Val
    ## LOC410509    -2.1502039 -0.6609070 -7.650419 1.583590e-11 1.661661e-07
    ## LOC412034    -1.8049836  3.0343159 -7.313710 8.196362e-11 3.997461e-07
    ## LOC102654070 -1.6041772 -0.3473674 -7.186070 1.142894e-10 3.997461e-07
    ## LOC551518     0.2338646  6.1667524  6.808487 7.035836e-10 1.845676e-06
    ## LOC413924    -1.0235549  1.9163459 -5.505481 2.925083e-07 6.138580e-04
    ## LOC102656292 -0.4616791  3.0676037 -5.017140 2.234900e-06 3.908468e-03
    ## LOC100578353 -0.8423844  0.6911799 -4.899199 3.642387e-06 5.459938e-03
    ## LOC107965375 -0.6417751  2.1393904 -4.699944 8.234560e-06 1.080065e-02
    ## LOC107965822 -0.6588857  5.1336058 -4.659473 1.037569e-05 1.209690e-02
    ## LOC412863    -0.3987873  5.9201781 -4.592690 1.328786e-05 1.338424e-02
    ## Mrjp3         3.0296128  0.4812802  4.583310 1.403094e-05 1.338424e-02
    ## LOC113219061 -0.5048024  2.4034193 -4.542533 1.533051e-05 1.340525e-02
    ## LOC107965692 -0.1459269  6.2832338 -4.514107 1.714472e-05 1.383842e-02
    ## LOC113218542  1.0406077  2.7427661  4.448916 2.346021e-05 1.758343e-02
    ## LOC409222     0.1234565  7.1555059  4.409351 2.579761e-05 1.804629e-02
    ## LOC725710    -0.7072772  2.8369133 -4.352184 3.349971e-05 2.166805e-02
    ## LOC551936    -0.2471613  3.7671536 -4.329379 3.510501e-05 2.166805e-02
    ## LOC551847    -0.3696546  3.6071100 -4.292414 4.067006e-05 2.370838e-02
    ## LOC552069    -0.4772645  1.3017701 -4.261489 4.547596e-05 2.484923e-02
    ## LOC102656664 -0.5025987  1.2696639 -4.250763 4.736343e-05 2.484923e-02
    ## LOC727293    -0.4420166  3.4666704 -4.120259 7.823734e-05 3.909259e-02
    ## Calpb        -0.1364310  8.0502586 -4.092471 8.568108e-05 4.086598e-02
    ## LOC100578804 -0.5963465  2.5333837 -4.052021 1.008818e-04 4.602404e-02
    ##                      B
    ## LOC410509    13.393295
    ## LOC412034    13.368783
    ## LOC102654070  7.981527
    ## LOC551518    12.260605
    ## LOC413924     5.778369
    ## LOC102656292  4.523409
    ## LOC100578353  2.923239
    ## LOC107965375  3.134186
    ## LOC107965822  3.101411
    ## LOC412863     2.839196
    ## Mrjp3         2.742287
    ## LOC113219061  2.737330
    ## LOC107965692  2.572391
    ## LOC113218542  2.433494
    ## LOC409222     2.153084
    ## LOC725710     2.143032
    ## LOC551936     2.122813
    ## LOC551847     1.985597
    ## LOC552069     1.171625
    ## LOC102656664  1.325392
    ## LOC727293     1.403903
    ## Calpb         1.000936
    ## LOC100578804  1.115086

### **CL_13 vs. CL**

``` r
CL_13vsCL_table
```

    ##                                                                                 description
    ## LOC102656492                                      cilia- and flagella-associated protein 44
    ## LOC100576157                                           uncharacterized protein LOC100576157
    ## aub                                                                               aubergine
    ## LOC113219349                                                   uncharacterized LOC113219349
    ## LOC100576828                                                      protein maelstrom homolog
    ## LOC107966034                                           uncharacterized protein LOC107966034
    ## LOC100576976                   putative leucine-rich repeat-containing protein DDB_G0290503
    ## LOC100578866                                                   uncharacterized LOC100578866
    ## Mrjp3                                                           major royal jelly protein 3
    ## LOC413697    PTS-dependent dihydroxyacetone kinase 1, dihydroxyacetone-binding subunit DhaK
    ##                   logFC    AveExpr         t      P.Value    adj.P.Val
    ## LOC102656492 -0.3290706  5.0349800 -6.054269 2.387119e-08 0.0002504804
    ## LOC100576157  0.6022160  3.3021060  5.685394 1.256560e-07 0.0006592544
    ## aub          -0.3803818  5.3448073 -5.319965 6.286281e-07 0.0021987314
    ## LOC113219349 -2.2509244 -0.7986247 -4.696922 8.638153e-06 0.0203472953
    ## LOC100576828  0.6106436  2.7425116  4.657916 9.695652e-06 0.0203472953
    ## LOC107966034  1.4170462 -2.0324201  4.583398 1.304410e-05 0.0228119634
    ## LOC100576976  0.4070028  3.2910845  4.524307 1.647102e-05 0.0246900578
    ## LOC100578866 -0.3410421  4.2144253 -4.474155 2.004918e-05 0.0262970013
    ## Mrjp3         3.3939827  0.4812802  4.301587 4.139748e-05 0.0482648620
    ## LOC413697     0.3585950  5.6581954  4.262722 4.654564e-05 0.0488403364
    ##                        B
    ## LOC102656492  8.88267021
    ## LOC100576157  6.95051044
    ## aub           5.82631428
    ## LOC113219349  0.04481661
    ## LOC100576828  2.88777463
    ## LOC107966034 -1.42797245
    ## LOC100576976  2.71186351
    ## LOC100578866  2.62815033
    ## Mrjp3         1.57161066
    ## LOC413697     1.78402397

# Export tables to Excel file

``` r
library(xlsx)
write.xlsx(MDvsCL_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="AcrossBrain_MDvsCL", row.names=TRUE)
write.xlsx(MDvsCL_13_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="AcrossBrain_MDvsCL_13", append=TRUE, row.names=TRUE)
write.xlsx(MDvsCLCL_13_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="AcrossBrain_MDvsCLCL_13", append=TRUE, row.names=TRUE)
write.xlsx(MDvsCL_Bifi_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="AcrossBrain_MDvsCL_Bifi", append=TRUE, row.names=TRUE)
write.xlsx(MDvsCLCL_13CL_Bifi_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="AcrossBrain_MDvsCLCL_13CL_Bifi", append=TRUE, row.names=TRUE)
# write.xlsx(CL_13vsCL_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="AcrossBrain_CL_13vsCL", append=TRUE, row.names=TRUE)
```

## **2) What are the effects of treatment on individual brain regions (the same contrasts as before but by brain region)?**

### **Optic lobes**

``` r
con <- makeContrasts(
    OL.MDvsCL= groupCLOL - groupMDOL,
    OL.MDvsCL_13= groupCL_13OL - groupMDOL,
    OL.MDvsCLCL_13= (groupCLOL + groupCL_13OL)/2 - groupMDOL,
    OL.MDvsCL_Bifi= groupCL_BifiOL - groupMDOL,
    OL.MDvsCLCL_13CL_Bifi= (groupCLOL + groupCL_13OL + groupCL_BifiOL)/3 - groupMDOL,
    OL.CL_13vsCL= groupCLOL - groupCL_13OL,
    levels=design)

fit2=contrasts.fit(fit,con)
fit2=eBayes(fit2, robust=TRUE)

summary(decideTests(fit2, p=0.05))

OL.MDvsCL_table=topTable(fit2,coef="OL.MDvsCL",n=Inf, sort="p", p=0.05)
OL.MDvsCL_13_table=topTable(fit2,coef="OL.MDvsCL_13",n=Inf, sort="p", p=0.05)
OL.MDvsCLCL_13_table=topTable(fit2,coef="OL.MDvsCLCL_13",n=Inf, sort="p", p=0.05)
OL.MDvsCL_Bifi_table=topTable(fit2,coef="OL.MDvsCL_Bifi",n=Inf, sort="p", p=0.05)
OL.MDvsCLCL_13CL_Bifi_table=topTable(fit2,coef="OL.MDvsCLCL_13CL_Bifi",n=Inf, sort="p", p=0.05)
OL.CL_13vsCL_table=topTable(fit2,coef="OL.CL_13vsCL",n=Inf, sort="p", p=0.05)
```

### **Optic lobes - MD vs. CL**

``` r
OL.MDvsCL_table
```

    ##                                                                  description
    ## LOC410509 uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC413924                           glyceraldehyde-3-phosphate dehydrogenase
    ##               logFC   AveExpr         t      P.Value  adj.P.Val        B
    ## LOC410509 -2.142635 -0.660907 -4.839578 4.990662e-06 0.04001425 2.741179
    ## LOC413924 -1.533410  1.916346 -4.725601 7.626846e-06 0.04001425 2.319194

### **Optic lobes - MD vs. CL_13**

``` r
OL.MDvsCL_13_table
```

    ##                                                                  description
    ## LOC410509 uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ##               logFC   AveExpr         t      P.Value    adj.P.Val        B
    ## LOC410509 -4.084794 -0.660907 -8.012585 2.740327e-12 2.875425e-08 10.54741

### **Optic lobes - MD vs. CL and CL_13**

``` r
OL.MDvsCLCL_13_table
```

    ##                                                                     description
    ## LOC410509    uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC413924                              glyceraldehyde-3-phosphate dehydrogenase
    ## LOC412034                                                     DNA helicase MCM8
    ## LOC102654070                                       uncharacterized LOC102654070
    ## LOC113218542                                       uncharacterized LOC113218542
    ## LOC412889                 branched-chain-amino-acid aminotransferase, cytosolic
    ##                   logFC    AveExpr         t      P.Value    adj.P.Val
    ## LOC410509    -3.1137147 -0.6609070 -7.779035 8.508662e-12 8.928139e-08
    ## LOC413924    -1.3995917  1.9163459 -5.172499 1.214477e-06 4.292188e-03
    ## LOC412034    -1.7953659  3.0343159 -5.183067 1.227158e-06 4.292188e-03
    ## LOC102654070 -1.6036272 -0.3473674 -4.724912 7.408504e-06 1.943436e-02
    ## LOC113218542  1.8435746  2.7427661  4.637785 1.124094e-05 2.359024e-02
    ## LOC412889    -0.4776883  4.2063486 -4.502229 1.808631e-05 3.162994e-02
    ##                       B
    ## LOC410509    11.6736000
    ## LOC413924     3.7124184
    ## LOC412034     4.5130714
    ## LOC102654070 -0.1292664
    ## LOC113218542  2.4098592
    ## LOC412889     2.7096953

### **Optic lobes - MD vs. CL_Bifi**

``` r
OL.MDvsCL_Bifi_table
```

    ## data frame with 0 columns and 0 rows

### **Optic lobes - MD vs. CL, CL_13 and CL_Bifi**

``` r
OL.MDvsCLCL_13CL_Bifi_table
```

    ##                                                                     description
    ## LOC410509    uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC413924                              glyceraldehyde-3-phosphate dehydrogenase
    ## LOC412034                                                     DNA helicase MCM8
    ## LOC102654070                                       uncharacterized LOC102654070
    ##                  logFC    AveExpr         t      P.Value    adj.P.Val
    ## LOC410509    -2.436172 -0.6609070 -6.470480 4.164443e-09 0.0000436975
    ## LOC413924    -1.357261  1.9163459 -5.402420 4.565683e-07 0.0017208977
    ## LOC412034    -1.776374  3.0343159 -5.399619 4.920131e-07 0.0017208977
    ## LOC102654070 -1.574207 -0.3473674 -4.946426 2.997593e-06 0.0078634366
    ##                       B
    ## LOC410509    7.18899099
    ## LOC413924    4.26891036
    ## LOC412034    5.09521913
    ## LOC102654070 0.03374164

### **Optic lobes - CL_13 vs. CL**

``` r
OL.CL_13vsCL_table
```

    ## data frame with 0 columns and 0 rows

### **Mushroom bodies**

``` r
con <- makeContrasts(
    MB.MDvsCL= groupCLMB - groupMDMB,
    MB.MDvsCL_13= groupCL_13MB - groupMDMB,
    MB.MDvsCLCL_13= (groupCLMB + groupCL_13MB)/2 - groupMDMB,
    MB.MDvsCL_Bifi= groupCL_BifiMB - groupMDMB,
    MB.MDvsCLCL_13CL_Bifi= (groupCLMB + groupCL_13MB + groupCL_BifiMB)/3 - groupMDMB,
    MB.CL_13vsCL= groupCLMB - groupCL_13MB,
    levels=design)

fit2=contrasts.fit(fit,con)
fit2=eBayes(fit2, robust=TRUE)

summary(decideTests(fit2, p=0.05))

MB.MDvsCL_table=topTable(fit2,coef="MB.MDvsCL",n=Inf, sort="p", p=0.05)
MB.MDvsCL_13_table=topTable(fit2,coef="MB.MDvsCL_13",n=Inf, sort="p", p=0.05)
MB.MDvsCLCL_13_table=topTable(fit2,coef="MB.MDvsCLCL_13",n=Inf, sort="p", p=0.05)
MB.MDvsCL_Bifi_table=topTable(fit2,coef="MB.MDvsCL_Bifi",n=Inf, sort="p", p=0.05)
MB.MDvsCLCL_13CL_Bifi_table=topTable(fit2,coef="MB.MDvsCLCL_13CL_Bifi",n=Inf, sort="p", p=0.05)
MB.CL_13vsCL_table=topTable(fit2,coef="MB.CL_13vsCL",n=Inf, sort="p", p=0.05)
```

### **Mushroom bodies - MD vs. CL**

``` r
MB.MDvsCL_table
```

    ## data frame with 0 columns and 0 rows

### **Mushroom bodies - MD vs. CL_13**

``` r
MB.MDvsCL_13_table
```

    ##                                                                     description
    ## LOC410509    uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC412034                                                     DNA helicase MCM8
    ## LOC100577359                                       uncharacterized LOC100577359
    ## LOC102654369                                               la-related protein 6
    ## LOC102654070                                       uncharacterized LOC102654070
    ##                   logFC    AveExpr         t      P.Value    adj.P.Val
    ## LOC410509    -3.4091659 -0.6609070 -6.148555 1.806659e-08 0.0001895727
    ## LOC412034    -2.1990194  3.0343159 -4.846130 4.898521e-06 0.0257000897
    ## LOC100577359  1.7208317  1.0544558  4.507084 1.822268e-05 0.0406111002
    ## LOC102654369 -0.3848238  6.1465990 -4.491302 1.886941e-05 0.0406111002
    ## LOC102654070 -1.8459145 -0.3473674 -4.483215 1.935152e-05 0.0406111002
    ##                      B
    ## LOC410509     4.892569
    ## LOC412034     2.931464
    ## LOC100577359  1.107095
    ## LOC102654369  2.666099
    ## LOC102654070 -1.049628

### **Mushroom bodies - MD vs. CL and CL_13**

``` r
MB.MDvsCLCL_13_table
```

    ##                                                                     description
    ## LOC410509    uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC102654070                                       uncharacterized LOC102654070
    ## LOC412034                                                     DNA helicase MCM8
    ## LOC107965375                                       uncharacterized LOC107965375
    ## LOC551518                                               glutamyl aminopeptidase
    ## LOC113219061                                       uncharacterized LOC113219061
    ##                   logFC    AveExpr         t      P.Value    adj.P.Val        B
    ## LOC410509    -2.5934204 -0.6609070 -5.878305 6.054644e-08 0.0006353138 5.719645
    ## LOC102654070 -1.8788592 -0.3473674 -5.383886 4.707829e-07 0.0024699626 1.549416
    ## LOC412034    -1.9897159  3.0343159 -5.135350 1.497220e-06 0.0052367760 4.351276
    ## LOC107965375 -0.7097531  2.1393904 -4.701492 8.183520e-06 0.0214674192 3.225351
    ## LOC551518     0.2324204  6.1667524  4.636253 1.057192e-05 0.0221862321 3.152889
    ## LOC113219061 -0.5729942  2.4034193 -4.443538 2.259131e-05 0.0395084388 2.419530

### **Mushroom bodies - MD vs. CL_Bifi**

``` r
MB.MDvsCL_Bifi_table
```

    ##                                   description     logFC    AveExpr         t
    ## LOC725217    uncharacterized protein PFB0145c -2.585646  1.9953126 -5.210109
    ## LOC102654070     uncharacterized LOC102654070 -2.063334 -0.3473674 -4.961665
    ##                   P.Value  adj.P.Val         B
    ## LOC725217    1.087643e-06 0.01141263  1.753348
    ## LOC102654070 2.814336e-06 0.01476541 -1.708404

### **Mushroom bodies - MD vs. CL, CL_13 and CL_Bifi**

``` r
MB.MDvsCLCL_13CL_Bifi_table
```

    ##                                                                     description
    ## LOC102654070                                       uncharacterized LOC102654070
    ## LOC412034                                                     DNA helicase MCM8
    ## LOC410509    uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC107965375                                       uncharacterized LOC107965375
    ## LOC113218535                                       uncharacterized LOC113218535
    ## LOC551518                                               glutamyl aminopeptidase
    ##                   logFC    AveExpr         t      P.Value    adj.P.Val        B
    ## LOC102654070 -1.9403508 -0.3473674 -5.947522 3.878244e-08 0.0004069442 2.434752
    ## LOC412034    -1.8855137  3.0343159 -5.218244 1.059161e-06 0.0055568878 4.547155
    ## LOC410509    -1.9574103 -0.6609070 -4.705160 8.553182e-06 0.0299161808 2.084912
    ## LOC107965375 -0.6476863  2.1393904 -4.551540 1.486446e-05 0.0383050267 2.668984
    ## LOC113218535 -0.9493515  0.6907690 -4.498146 1.825266e-05 0.0383050267 1.110154
    ## LOC551518     0.2108724  6.1667524  4.423806 2.439165e-05 0.0426569311 2.393644

### **Mushroom bodies - CL_13 vs. CL**

``` r
MB.CL_13vsCL_table
```

    ## data frame with 0 columns and 0 rows

### **Antennal lobes + subaesophageal ganglion**

``` r
con <- makeContrasts(
    AL.MDvsCL= groupCLAL - groupMDAL,
    AL.MDvsCL_13= groupCL_13AL - groupMDAL,
    AL.MDvsCLCL_13= (groupCLAL + groupCL_13AL)/2 - groupMDAL,
    AL.MDvsCL_Bifi= groupCL_BifiAL - groupMDAL,
    AL.MDvsCLCL_13CL_Bifi= (groupCLAL + groupCL_13AL + groupCL_BifiAL)/3 - groupMDAL,
    AL.CL_13vsCL= groupCLAL - groupCL_13AL,
    levels=design)

fit2=contrasts.fit(fit,con)
fit2=eBayes(fit2, robust=TRUE)

summary(decideTests(fit2, p=0.05))

AL.MDvsCL_table=topTable(fit2,coef="AL.MDvsCL",n=Inf, sort="p", p=0.05)
AL.MDvsCL_13_table=topTable(fit2,coef="AL.MDvsCL_13",n=Inf, sort="p", p=0.05)
AL.MDvsCLCL_13_table=topTable(fit2,coef="AL.MDvsCLCL_13",n=Inf, sort="p", p=0.05)
AL.MDvsCL_Bifi_table=topTable(fit2,coef="AL.MDvsCL_Bifi",n=Inf, sort="p", p=0.05)
AL.MDvsCLCL_13CL_Bifi_table=topTable(fit2,coef="AL.MDvsCLCL_13CL_Bifi",n=Inf, sort="p", p=0.05)
AL.CL_13vsCL_table=topTable(fit2,coef="AL.CL_13vsCL",n=Inf, sort="p", p=0.05)
```

### **Antennal lobes + subaesophageal ganglion - MD vs. CL**

``` r
AL.MDvsCL_table
```

    ##                                                                     description
    ## Mrjp3                                               major royal jelly protein 3
    ## LOC551518                                               glutamyl aminopeptidase
    ## LOC410509    uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC409578                                                  adapter molecule Crk
    ## Mrjp1                                               major royal jelly protein 1
    ## LOC411931                                      ELMO domain-containing protein C
    ## Mrjp2                                               major royal jelly protein 2
    ## LOC551109                                           kinesin-like protein Klp61F
    ## LOC409222                                                              cyclin G
    ## LOC100576747                                             maternal protein tudor
    ## LOC412832                       structural maintenance of chromosomes protein 2
    ## LOC102654070                                       uncharacterized LOC102654070
    ## Mrjp4                                               major royal jelly protein 4
    ## LOC552663                                                      pyridoxal kinase
    ## LOC113219378                                        large subunit ribosomal RNA
    ##                   logFC     AveExpr         t      P.Value    adj.P.Val
    ## Mrjp3         7.4824983  0.48128025  6.802917 9.353925e-10 9.815074e-06
    ## LOC551518     0.3543717  6.16675241  6.139433 1.616569e-08 8.481331e-05
    ## LOC410509    -2.1041438 -0.66090696 -5.114949 1.614477e-06 3.630686e-03
    ## LOC409578     0.2330026  5.79178111  5.082498 1.700355e-06 3.630686e-03
    ## Mrjp1         6.6844745  3.20457997  5.103124 1.730052e-06 3.630686e-03
    ## LOC411931    -0.2506028  6.30823850 -4.959059 2.844876e-06 4.975214e-03
    ## Mrjp2         4.5731310  2.16075218  4.779112 6.468694e-06 9.696572e-03
    ## LOC551109    -0.4389372  4.50165494 -4.569977 1.375596e-05 1.804266e-02
    ## LOC409222     0.2111754  7.15550592  4.486146 1.913088e-05 2.230448e-02
    ## LOC100576747 -0.3192734  6.40216091 -4.422326 2.453210e-05 2.574153e-02
    ## LOC412832    -0.4114831  4.81227910 -4.345954 3.294279e-05 3.104138e-02
    ## LOC102654070 -1.4833000 -0.34736738 -4.326461 3.549953e-05 3.104138e-02
    ## Mrjp4         4.3747340  0.24951154  4.263385 4.777039e-05 3.855805e-02
    ## LOC552663    -0.3156118  5.23623411 -4.169429 6.433922e-05 4.582296e-02
    ## LOC113219378  1.8552885 -0.06248023  4.166001 6.550504e-05 4.582296e-02
    ##                       B
    ## Mrjp3        10.2120769
    ## LOC551518     9.2753618
    ## LOC410509     4.1569052
    ## LOC409578     4.8725866
    ## Mrjp1         4.8636261
    ## LOC411931     4.3710114
    ## Mrjp2         3.6196748
    ## LOC551109     2.9732487
    ## LOC409222     2.5432051
    ## LOC100576747  2.3325224
    ## LOC412832     2.1717847
    ## LOC102654070  0.2161527
    ## Mrjp4         1.0456268
    ## LOC552663     1.4933758
    ## LOC113219378 -0.3565333

### **Antennal lobes + subaesophageal ganglion - MD vs. CL_13**

``` r
AL.MDvsCL_13_table
```

    ##                                                                     description
    ## LOC410509    uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC100577359                                       uncharacterized LOC100577359
    ## LOC412034                                                     DNA helicase MCM8
    ## LOC551518                                               glutamyl aminopeptidase
    ## LOC107965692                                    poly(ADP-ribose) glycohydrolase
    ## LOC408371                             probable ATP-dependent RNA helicase DDX56
    ##                   logFC   AveExpr         t      P.Value   adj.P.Val        B
    ## LOC410509    -2.5173604 -0.660907 -5.579096 2.248264e-07 0.002174202 4.852346
    ## LOC100577359  1.7947751  1.054456  5.427613 4.144100e-07 0.002174202 4.487988
    ## LOC412034    -2.0237922  3.034316 -4.952935 3.175184e-06 0.011105735 3.705248
    ## LOC551518     0.2784276  6.166752  4.820532 5.027282e-06 0.013187818 3.881450
    ## LOC107965692 -0.2489586  6.283234 -4.563773 1.409761e-05 0.029585249 2.914610
    ## LOC408371    -0.2718077  4.443522 -4.500583 1.807912e-05 0.031617371 2.715029

### **Antennal lobes + subaesophageal ganglion - MD vs. CL and CL_13**

``` r
AL.MDvsCLCL_13_table
```

    ##                                                                     description
    ## LOC410509    uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC551518                                               glutamyl aminopeptidase
    ## LOC409578                                                  adapter molecule Crk
    ## LOC412034                                                     DNA helicase MCM8
    ## Mrjp3                                               major royal jelly protein 3
    ## LOC100577359                                       uncharacterized LOC100577359
    ## LOC409222                                                              cyclin G
    ## LOC551109                                           kinesin-like protein Klp61F
    ## Mrjp1                                               major royal jelly protein 1
    ## LOC100576747                                             maternal protein tudor
    ## LOC107965692                                    poly(ADP-ribose) glycohydrolase
    ## LOC102654070                                       uncharacterized LOC102654070
    ## LOC113218542                                       uncharacterized LOC113218542
    ## LOC409164                                             uncharacterized LOC409164
    ## LOC408371                             probable ATP-dependent RNA helicase DDX56
    ## LOC551718                                             ubiquitin-like protein 4A
    ## LOC724188                          protein transport protein Sec61 subunit beta
    ## LOC113219359                                          autophagy protein 12-like
    ## LOC724487                           28 kDa heat- and acid-stable phosphoprotein
    ## LOC107965146                                       uncharacterized LOC107965146
    ## LOC411931                                      ELMO domain-containing protein C
    ## LOC102654365       glutamyl-tRNA(Gln) amidotransferase subunit B, mitochondrial
    ## LOC724659                                   ribonuclease P protein subunit rpr2
    ## LOC113219283            ATP synthase mitochondrial F1 complex assembly factor 1
    ## LOC408439                       MATH and LRR domain-containing protein PFE0570w
    ## LOC552663                                                      pyridoxal kinase
    ## LOC726721                         protein transport protein Sec61 subunit gamma
    ##                   logFC    AveExpr         t      P.Value    adj.P.Val
    ## LOC410509    -2.3107521 -0.6609070 -6.353169 7.131425e-09 5.233348e-05
    ## LOC551518     0.3163997  6.1667524  6.244254 9.974932e-09 5.233348e-05
    ## LOC409578     0.2131922  5.7917811  5.297856 6.818563e-07 2.384906e-03
    ## LOC412034    -1.7949267  3.0343159 -5.105237 1.696663e-06 4.450771e-03
    ## Mrjp3         5.1068167  0.4812802  4.926685 3.566965e-06 7.485633e-03
    ## LOC100577359  1.3572910  1.0544558  4.536889 1.621990e-05 1.694390e-02
    ## LOC409222     0.1853741  7.1555059  4.499304 1.817000e-05 1.694390e-02
    ## LOC551109    -0.3720083  4.5016549 -4.493283 1.860372e-05 1.694390e-02
    ## Mrjp1         5.3694603  3.2045800  4.508984 1.874028e-05 1.694390e-02
    ## LOC100576747 -0.2822704  6.4021609 -4.483917 1.929844e-05 1.694390e-02
    ## LOC107965692 -0.2128705  6.2832338 -4.470322 2.035158e-05 1.694390e-02
    ## LOC102654070 -1.2953565 -0.3473674 -4.459712 2.121191e-05 1.694390e-02
    ## LOC113218542  1.6537351  2.7427661  4.465349 2.202073e-05 1.694390e-02
    ## LOC409164     0.2017593  7.3278188  4.443360 2.260693e-05 1.694390e-02
    ## LOC408371    -0.2311031  4.4435221 -4.405748 2.616005e-05 1.829982e-02
    ## LOC551718     0.2284829  4.7612657  4.285375 4.152921e-05 2.723537e-02
    ## LOC724188     0.2168090  5.5032536  4.264980 4.487733e-05 2.769987e-02
    ## LOC113219359 -0.2269083  5.7508345 -4.155256 6.784155e-05 3.887393e-02
    ## LOC724487    -0.2536941  5.1673030 -4.132525 7.384386e-05 3.887393e-02
    ## LOC107965146  0.3305066  4.4841662  4.120146 7.732364e-05 3.887393e-02
    ## LOC411931    -0.1811798  6.3082385 -4.118494 7.779972e-05 3.887393e-02
    ## LOC102654365 -0.2257254  5.8263338 -4.065653 9.460142e-05 4.312332e-02
    ## LOC724659    -0.4952567  2.0427836 -4.059781 9.667005e-05 4.312332e-02
    ## LOC113219283 -0.3284928  4.6455688 -4.054319 9.863334e-05 4.312332e-02
    ## LOC408439    -0.2515978  5.5138211 -4.036942 1.051358e-04 4.412759e-02
    ## LOC552663    -0.2628067  5.2362341 -3.994237 1.229063e-04 4.928930e-02
    ## LOC726721     0.2777599  4.9467238  3.985612 1.268285e-04 4.928930e-02
    ##                      B
    ## LOC410509    7.9249198
    ## LOC551518    9.7252908
    ## LOC409578    5.7486909
    ## LOC412034    4.3291238
    ## Mrjp3        3.1879414
    ## LOC100577359 1.8643572
    ## LOC409222    2.6424436
    ## LOC551109    2.6913477
    ## Mrjp1        2.6681216
    ## LOC100576747 2.6029064
    ## LOC107965692 2.5575293
    ## LOC102654070 0.2444089
    ## LOC113218542 2.1238722
    ## LOC409164    2.4365264
    ## LOC408371    2.3880695
    ## LOC551718    1.9611100
    ## LOC724188    1.8624749
    ## LOC113219359 1.4685366
    ## LOC724487    1.4081719
    ## LOC107965146 1.4087774
    ## LOC411931    1.3168140
    ## LOC102654365 1.1596719
    ## LOC724659    0.5251067
    ## LOC113219283 1.1845489
    ## LOC408439    1.0743825
    ## LOC552663    0.9378933
    ## LOC726721    0.9343625

### **Antennal lobes + subaesophageal ganglion - MD vs. CL_Bifi**

``` r
AL.MDvsCL_Bifi_table
```

    ##                                               description      logFC   AveExpr
    ## Mrjp3                         major royal jelly protein 3  7.4194471 0.4812802
    ## LOC551718                       ubiquitin-like protein 4A  0.3423207 4.7612657
    ## LOC551518                         glutamyl aminopeptidase  0.3249414 6.1667524
    ## LOC725217                uncharacterized protein PFB0145c -2.3885397 1.9953126
    ## LOC409202                        40S ribosomal protein S9  0.2491057 6.5317263
    ## LOC413021                                         papilin  0.5406685 6.9752223
    ## LOC726126                              proclotting enzyme  0.7588542 4.4766429
    ## Mrjp2                         major royal jelly protein 2  4.3026879 2.1607522
    ## LOC409578                            adapter molecule Crk  0.2065573 5.7917811
    ## LOC410225    probable E3 ubiquitin-protein ligase RNF144A  0.3400645 5.4796217
    ## Mrjp1                         major royal jelly protein 1  5.6455440 3.2045800
    ## LOC410120                      thioredoxin, mitochondrial  0.3643176 4.8602305
    ## Mrjp4                         major royal jelly protein 4  4.3113227 0.2495115
    ## LOC102654507                 uncharacterized LOC102654507 -0.6229738 3.2555583
    ##                      t      P.Value    adj.P.Val         B
    ## Mrjp3         6.762059 1.131147e-09 1.186912e-05 9.5135185
    ## LOC551718     5.657295 1.422972e-07 5.765460e-04 7.1828816
    ## LOC551518     5.624040 1.648373e-07 5.765460e-04 7.0803224
    ## LOC725217    -4.887006 4.126657e-06 1.082525e-02 2.7224729
    ## LOC409202     4.772694 6.106844e-06 1.153635e-02 3.6798672
    ## LOC413021     4.749969 7.007980e-06 1.153635e-02 3.5468452
    ## LOC726126     4.725346 7.696030e-06 1.153635e-02 3.4895896
    ## Mrjp2         4.557214 1.553705e-05 2.037878e-02 2.7893846
    ## LOC409578     4.493160 1.861273e-05 2.044006e-02 2.6692234
    ## LOC410225     4.481527 1.947971e-05 2.044006e-02 2.6474033
    ## Mrjp1         4.385315 3.014264e-05 2.875334e-02 2.2567584
    ## LOC410120     4.269647 4.408912e-05 3.855226e-02 1.9070311
    ## Mrjp4         4.242349 5.168605e-05 4.171859e-02 0.7331008
    ## LOC102654507 -4.200765 5.720170e-05 4.287268e-02 1.5585382

### **Antennal lobes + subaesophageal ganglion - MD vs. CL, CL_13 and CL_Bifi**

``` r
AL.MDvsCLCL_13CL_Bifi_table
```

    ##                                                                     description
    ## LOC551518                                               glutamyl aminopeptidase
    ## Mrjp3                                               major royal jelly protein 3
    ## LOC410509    uncharacterized threonine-rich GPI-anchored glycoprotein PJ4664.02
    ## LOC409578                                                  adapter molecule Crk
    ## LOC412034                                                     DNA helicase MCM8
    ## LOC551718                                             ubiquitin-like protein 4A
    ## LOC102654070                                       uncharacterized LOC102654070
    ## Mrjp1                                               major royal jelly protein 1
    ## LOC409222                                                              cyclin G
    ## LOC409164                                             uncharacterized LOC409164
    ## LOC551109                                           kinesin-like protein Klp61F
    ## LOC410225                          probable E3 ubiquitin-protein ligase RNF144A
    ## LOC408371                             probable ATP-dependent RNA helicase DDX56
    ## LOC410120                                            thioredoxin, mitochondrial
    ## LOC409366                                          integral membrane protein 2C
    ## LOC107965692                                    poly(ADP-ribose) glycohydrolase
    ## LOC724487                           28 kDa heat- and acid-stable phosphoprotein
    ## LOC100577359                                       uncharacterized LOC100577359
    ## Mrjp2                                               major royal jelly protein 2
    ## LOC411931                                      ELMO domain-containing protein C
    ## LOC107965146                                       uncharacterized LOC107965146
    ## LOC100576747                                             maternal protein tudor
    ## LOC107965375                                       uncharacterized LOC107965375
    ## LOC408501                 serine/threonine-protein kinase/endoribonuclease IRE1
    ## LOC102654507                                       uncharacterized LOC102654507
    ## LOC726721                         protein transport protein Sec61 subunit gamma
    ## LOC412554                     transmembrane emp24 domain-containing protein eca
    ## LOC413021                                                               papilin
    ## LOC100576848                               uncharacterized protein LOC100576848
    ## LOC724188                          protein transport protein Sec61 subunit beta
    ## LOC409202                                              40S ribosomal protein S9
    ## LOC107964187                                       uncharacterized LOC107964187
    ## LOC113219359                                          autophagy protein 12-like
    ## LOC552663                                                      pyridoxal kinase
    ## LOC725980                              28S ribosomal protein S36, mitochondrial
    ##                   logFC    AveExpr         t      P.Value    adj.P.Val
    ## LOC551518     0.3192469  6.1667524  6.625079 1.681186e-09 1.764069e-05
    ## Mrjp3         5.8776935  0.4812802  5.968921 4.180738e-08 2.193424e-04
    ## LOC410509    -2.0570291 -0.6609070 -5.831966 7.433067e-08 2.599839e-04
    ## LOC409578     0.2109805  5.7917811  5.512710 2.688650e-07 7.053000e-04
    ## LOC412034    -1.7530635  3.0343159 -5.297871 7.576195e-07 1.443661e-03
    ## LOC551718     0.2664288  4.7612657  5.253179 8.254993e-07 1.443661e-03
    ## LOC102654070 -1.2979741 -0.3473674 -4.763029 6.350823e-06 8.825323e-03
    ## Mrjp1         5.4614882  3.2045800  4.769280 6.728541e-06 8.825323e-03
    ## LOC409222     0.1829699  7.1555059  4.674177 9.084553e-06 1.059158e-02
    ## LOC409164     0.1987544  7.3278188  4.606502 1.190138e-05 1.248812e-02
    ## LOC551109    -0.3580713  4.5016549 -4.578595 1.329466e-05 1.268189e-02
    ## LOC410225     0.2831395  5.4796217  4.440607 2.285032e-05 1.998070e-02
    ## LOC408371    -0.2133400  4.4435221 -4.298285 3.953562e-05 3.088961e-02
    ## LOC410120     0.3058713  4.8602305  4.280908 4.224134e-05 3.088961e-02
    ## LOC409366     0.1369636  8.7312287  4.269239 4.415746e-05 3.088961e-02
    ## LOC107965692 -0.1916873  6.2832338 -4.243260 4.872834e-05 3.195665e-02
    ## LOC724487    -0.2447922  5.1673030 -4.204052 5.649865e-05 3.338624e-02
    ## LOC100577359  1.2047729  1.0544558  4.198965 5.912424e-05 3.338624e-02
    ## Mrjp2         3.5044730  2.1607522  4.187446 6.344812e-05 3.338624e-02
    ## LOC411931    -0.1741783  6.3082385 -4.172367 6.363526e-05 3.338624e-02
    ## LOC107965146  0.3176719  4.4841662  4.154211 6.810694e-05 3.373591e-02
    ## LOC100576747 -0.2474864  6.4021609 -4.144079 7.073191e-05 3.373591e-02
    ## LOC107965375 -0.7943990  2.1393904 -4.064327 9.536671e-05 4.153282e-02
    ## LOC408501    -0.1929906  6.8504765 -4.051335 9.972189e-05 4.153282e-02
    ## LOC102654507 -0.4866197  3.2555583 -4.044314 1.023287e-04 4.153282e-02
    ## LOC726721     0.2680981  4.9467238  4.042767 1.029118e-04 4.153282e-02
    ## LOC412554     0.2037955  5.4502087  4.031676 1.071862e-04 4.165574e-02
    ## LOC413021     0.3799940  6.9752223  4.007329 1.202208e-04 4.387867e-02
    ## LOC100576848  0.2758926  4.1697289  3.989935 1.248479e-04 4.387867e-02
    ## LOC724188     0.1928978  5.5032536  3.988612 1.254513e-04 4.387867e-02
    ## LOC409202     0.1728532  6.5317263  3.979231 1.298072e-04 4.393763e-02
    ## LOC107964187 -0.8058571  0.6691218 -3.964622 1.368802e-04 4.488386e-02
    ## LOC113219359 -0.2038658  5.7508345 -3.939233 1.500577e-04 4.682316e-02
    ## LOC552663    -0.2453384  5.2362341 -3.934638 1.525686e-04 4.682316e-02
    ## LOC725980    -0.2886505  3.7271433 -3.928151 1.561813e-04 4.682316e-02
    ##                        B
    ## LOC551518    11.42885174
    ## Mrjp3         6.87056590
    ## LOC410509     6.43400614
    ## LOC409578     6.61556359
    ## LOC412034     5.18722950
    ## LOC551718     5.57082550
    ## LOC102654070  1.32877757
    ## Mrjp1         3.61098097
    ## LOC409222     3.25208041
    ## LOC409164     2.99583093
    ## LOC551109     3.00452869
    ## LOC410225     2.47796677
    ## LOC408371     2.00904674
    ## LOC410120     1.92321623
    ## LOC409366     1.73745171
    ## LOC107965692  1.70008712
    ## LOC724487     1.61933482
    ## LOC100577359  1.08684296
    ## Mrjp2         1.57425118
    ## LOC411931     1.45784731
    ## LOC107965146  1.52660250
    ## LOC100576747  1.34645809
    ## LOC107965375  0.85805159
    ## LOC408501     1.01031804
    ## LOC102654507  1.13244534
    ## LOC726721     1.09348932
    ## LOC412554     1.02333122
    ## LOC413021     0.83693933
    ## LOC100576848  0.98619432
    ## LOC724188     0.87760928
    ## LOC409202     0.77657361
    ## LOC107964187 -0.02254017
    ## LOC113219359  0.69192840
    ## LOC552663     0.69734428
    ## LOC725980     0.78725791

### **Antennal lobes + subaesophageal ganglion - CL_13 vs. CL**

``` r
AL.CL_13vsCL_table
```

    ## data frame with 0 columns and 0 rows

# Export region-specific DEG tables to Excel file (some comparisons had 0 DEGs so we do not export them)

``` r
library(xlsx)
write.xlsx(AL.MDvsCL_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="AL.MDvsCL", append=TRUE, row.names=TRUE)
write.xlsx(AL.MDvsCL_13_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="AL.MDvsCL_13", append=TRUE, row.names=TRUE)
write.xlsx(AL.MDvsCLCL_13_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="AL.MDvsCLCL_13_table", append=TRUE, row.names=TRUE)
write.xlsx(AL.MDvsCL_Bifi_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="AL.MDvsCL_Bifi", append=TRUE, row.names=TRUE)
write.xlsx(AL.MDvsCLCL_13CL_Bifi_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="AL.MDvsCLCL_13CL_Bifi", append=TRUE, row.names=TRUE)
# write.xlsx(AL.CL_13vsCL_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="AL.CL_13vsCL", append=TRUE, row.names=TRUE)

# write.xlsx(MB.MDvsCL_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="MB.MDvsCL", append=TRUE, row.names=TRUE)
write.xlsx(MB.MDvsCL_13_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="MB.MDvsCL_13", append=TRUE, row.names=TRUE)
write.xlsx(MB.MDvsCLCL_13_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="MB.MDvsCLCL_13_table", append=TRUE, row.names=TRUE)
write.xlsx(MB.MDvsCL_Bifi_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="MB.MDvsCL_Bifi", append=TRUE, row.names=TRUE)
write.xlsx(MB.MDvsCLCL_13CL_Bifi_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="MB.MDvsCLCL_13CL_Bifi", append=TRUE, row.names=TRUE)
# write.xlsx(MB.CL_13vsCL_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="MB.CL_13vsCL", append=TRUE, row.names=TRUE)

write.xlsx(OL.MDvsCL_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="OL.MDvsCL", append=TRUE, row.names=TRUE)
write.xlsx(OL.MDvsCL_13_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="OL.MDvsCL_13", append=TRUE, row.names=TRUE)
write.xlsx(OL.MDvsCLCL_13_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="OL.MDvsCLCL_13_table", append=TRUE, row.names=TRUE)
# write.xlsx(OL.MDvsCL_Bifi_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="OL.MDvsCL_Bifi", append=TRUE, row.names=TRUE)
write.xlsx(OL.MDvsCLCL_13CL_Bifi_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="OL.MDvsCLCL_13CL_Bifi", append=TRUE, row.names=TRUE)
# write.xlsx(OL.CL_13vsCL_table, file="~/Desktop/Brain_DEGresults.xlsx", sheetName="OL.CL_13vsCL", append=TRUE, row.names=TRUE)
```

## Store DEG lists

``` r
AL.MDvsCL_DEGs <- row.names(AL.MDvsCL_table)
AL.MDvsCL_13_DEGs <- row.names(AL.MDvsCL_13_table)
AL.MDvsCLCL_13_DEGs <- row.names(AL.MDvsCLCL_13_table)
AL.MDvsCL_Bifi_DEGs <- row.names(AL.MDvsCL_Bifi_table)
AL.MDvsCLCL_13CL_Bifi_DEGs <- row.names(AL.MDvsCLCL_13CL_Bifi_table)
AL.CL_13vsCL_DEGs <- row.names(AL.CL_13vsCL_table)
MB.MDvsCL_DEGs <- row.names(MB.MDvsCL_table)
MB.MDvsCL_13_DEGs <- row.names(MB.MDvsCL_13_table)
MB.MDvsCLCL_13_DEGs <- row.names(MB.MDvsCLCL_13_table)
MB.MDvsCL_Bifi_DEGs <- row.names(MB.MDvsCL_Bifi_table)
MB.MDvsCLCL_13CL_Bifi_DEGs <- row.names(MB.MDvsCLCL_13CL_Bifi_table)
MB.CL_13vsCL_DEGs <- row.names(MB.CL_13vsCL_table)
OL.MDvsCL_DEGs <- row.names(OL.MDvsCL_table)
OL.MDvsCL_13_DEGs <- row.names(OL.MDvsCL_13_table)
OL.MDvsCLCL_13_DEGs <- row.names(OL.MDvsCLCL_13_table)
OL.MDvsCL_Bifi_DEGs <- row.names(OL.MDvsCL_Bifi_table)
OL.MDvsCLCL_13CL_Bifi_DEGs <- row.names(OL.MDvsCLCL_13CL_Bifi_table)
OL.CL_13vsCL_DEGs <- row.names(OL.CL_13vsCL_table)

AllDEGs <- unique(sort(c(MDvsCL_DEGs, MDvsCL_13_DEGs, MDvsCLCL_13_DEGs, MDvsCL_Bifi_DEGs, MDvsCLCL_13CL_Bifi_DEGs, AL.MDvsCL_DEGs, AL.MDvsCL_13_DEGs, AL.MDvsCLCL_13_DEGs, AL.MDvsCL_Bifi_DEGs, AL.MDvsCLCL_13CL_Bifi_DEGs, MB.MDvsCL_DEGs, MB.MDvsCL_13_DEGs, MB.MDvsCLCL_13_DEGs, MB.MDvsCL_Bifi_DEGs, MB.MDvsCLCL_13CL_Bifi_DEGs, OL.MDvsCL_DEGs, OL.MDvsCL_13_DEGs, OL.MDvsCLCL_13_DEGs, OL.MDvsCL_Bifi_DEGs, OL.MDvsCLCL_13CL_Bifi_DEGs)))    

AllDEGswCLvsCL_13 <- unique(sort(c(MDvsCL_DEGs, MDvsCL_13_DEGs, MDvsCLCL_13_DEGs, MDvsCL_Bifi_DEGs, MDvsCLCL_13CL_Bifi_DEGs, AL.MDvsCL_DEGs, AL.MDvsCL_13_DEGs, AL.MDvsCLCL_13_DEGs, AL.MDvsCL_Bifi_DEGs, AL.MDvsCLCL_13CL_Bifi_DEGs, MB.MDvsCL_DEGs, MB.MDvsCL_13_DEGs, MB.MDvsCLCL_13_DEGs, MB.MDvsCL_Bifi_DEGs, MB.MDvsCLCL_13CL_Bifi_DEGs, OL.MDvsCL_DEGs, OL.MDvsCL_13_DEGs, OL.MDvsCLCL_13_DEGs, OL.MDvsCL_Bifi_DEGs, OL.MDvsCLCL_13CL_Bifi_DEGs, CL_13vsCL_DEGs, AL.CL_13vsCL_DEGs, MB.CL_13vsCL_DEGs, OL.CL_13vsCL_DEGs)))

All_AL_DEGs <- unique(sort(c(AL.MDvsCL_DEGs, AL.MDvsCL_13_DEGs, AL.MDvsCLCL_13_DEGs, AL.MDvsCL_Bifi_DEGs, AL.MDvsCLCL_13CL_Bifi_DEGs))) 
All_MB_DEGs <- unique(sort(c(MB.MDvsCL_DEGs, MB.MDvsCL_13_DEGs, MB.MDvsCLCL_13_DEGs, MB.MDvsCL_Bifi_DEGs, MB.MDvsCLCL_13CL_Bifi_DEGs))) 
All_OL_DEGs <- unique(sort(c(OL.MDvsCL_DEGs, OL.MDvsCL_13_DEGs, OL.MDvsCLCL_13_DEGs, OL.MDvsCL_Bifi_DEGs, OL.MDvsCLCL_13CL_Bifi_DEGs))) 

All_MDvsCL <- unique(sort(c(MDvsCL_DEGs, AL.MDvsCL_DEGs, MB.MDvsCL_DEGs, OL.MDvsCL_DEGs)))
All_MDvsCL_13 <- unique(sort(c(MDvsCL_13_DEGs, AL.MDvsCL_13_DEGs, MB.MDvsCL_13_DEGs, OL.MDvsCL_13_DEGs))) 
All_MDvsCL_Bifi <- unique(sort(c(MDvsCL_Bifi_DEGs, AL.MDvsCL_Bifi_DEGs, MB.MDvsCL_Bifi_DEGs, OL.MDvsCL_Bifi_DEGs))) 
```

## Plot ordinations of all identified DEGs

``` r
library(PCAtools)

lcpmAllDEGs<-lcpm[row.names(lcpm)%in%AllDEGs,] 

#row.names(Design) <- Design[,1]
row.names(Design_AL) <- Design_AL[,1]
row.names(Design_MB) <- Design_MB[,1]
row.names(Design_OL) <- Design_OL[,1]

AL.lcpmAllDEGs<-lcpmAllDEGs[,colnames(lcpmAllDEGs)%in%Design_AL[,1]] 
MB.lcpmAllDEGs<-lcpmAllDEGs[,colnames(lcpmAllDEGs)%in%Design_MB[,1]] 
OL.lcpmAllDEGs<-lcpmAllDEGs[,colnames(lcpmAllDEGs)%in%Design_OL[,1]] 

p1 <- pca(AL.lcpmAllDEGs, metadata = Design_AL)
p2 <- pca(MB.lcpmAllDEGs, metadata = Design_MB)
p3 <- pca(OL.lcpmAllDEGs, metadata = Design_OL)

pl1 <- biplot(p1, x = 'PC1', y = 'PC2',  lab = NULL,
    colby = 'Treatment', 
    hline = 0, vline = 0,
    legendPosition = 'right') +
    scale_color_manual(values = c("#00BFC4","#7CAE00","#F8766D","#C77CFF"))

pl2 <- biplot(p2, x = 'PC1', y = 'PC2', lab = NULL,
    colby = 'Treatment', 
    hline = 0, vline = 0,
    legendPosition = 'right') +
    scale_color_manual(values = c("#00BFC4","#7CAE00","#F8766D","#C77CFF"))

pl3 <- biplot(p3, x = 'PC1', y = 'PC2',  lab = NULL,
    colby = 'Treatment', 
    hline = 0, vline = 0,
    legendPosition = 'right') +
    scale_color_manual(values = c("#00BFC4","#7CAE00","#F8766D","#C77CFF"))

library(ggpubr)
ggarrange(pl1,pl2,pl3, nrow=1, common.legend = T)
```

![](02-Brain-RNAseqAnalysis_files/figure-gfm/unnamed-chunk-43-1.png)<!-- -->

``` r
# ggsave(file="~/Desktop/PCA-RegionSpecific-DEGs.pdf", width=10, height=5, useDingbats=FALSE)
```

## **Venn diagrams to compare overlap in DEG lists**

``` r
gutDEGs <- read.table( "~/Desktop/Gut_allDEGs.txt", sep="\t", header = F)
gutDEGs <- gutDEGs$V1

# Make a list containing all DEG comparison lists
DEGs_gut_brain <- list(gutDEGs, AllDEGs)
DEGs_Treat_BrainRegions <- list(All_AL_DEGs, All_MB_DEGs, All_OL_DEGs)
DEGs_3wayComparisons <- list(All_MDvsCL, All_MDvsCL_13, All_MDvsCL_Bifi) 

# We can rename our list vectors
names(DEGs_gut_brain) <- c("Gut", "Brain")
names(DEGs_Treat_BrainRegions) <- c("AL", "MB", "OL")
names(DEGs_3wayComparisons) <- c("MD vs CL", "MD vs CL_13", "MD vs CL_Bifi")

# Now we can plot a Venn diagram with the VennDiagram R package, as follows:
require("VennDiagram")

overrideTriple=T
venn.plot1 <- venn.diagram(DEGs_gut_brain, NULL, euler.d = F, scaled = F, fill=c("darkmagenta", "darkblue"), alpha=c(0.5,0.5), cex = 1.6, cat.fontface=4, category.names=c("Gut", "Brain"), main="Differentially expressed genes in gut and brain by gut microbiota")

overrideTriple=T
venn.plot2 <- venn.diagram(DEGs_Treat_BrainRegions, NULL, euler.d = F, scaled = F, fill=c("darkmagenta", "darkblue", "green"), alpha=c(0.5,0.5,0.5), cex = 1.6, cat.fontface=4, category.names=c("AL", "MB", "OL"), main="Differentially expressed genes in brain by gut microbiota")

overrideTriple=T
venn.plot3 <- venn.diagram(DEGs_3wayComparisons, NULL, euler.d = F, scaled = F, fill=c("darkmagenta", "darkblue", "green"), alpha=c(0.5,0.5,0.5), cex = 1.6, cat.fontface=4, category.names=c("MD vs CL", "MD vs CL_13", "MD vs CL_Bifi"), main="Differentially expressed genes in the brain")


grid.newpage()
# dev.new()
# pdf("~/Desktop/Venn_GutMicrobiota_GutVsBrain.pdf",height=4,width=6)  
p1 <- grid.draw(venn.plot1)
```

![](02-Brain-RNAseqAnalysis_files/figure-gfm/unnamed-chunk-44-1.png)<!-- -->

``` r
# dev.off()

grid.newpage()
# dev.new()
# pdf("~/Desktop/Venn_GutMicrobiota_BrainRegions.pdf",height=4,width=6)  
p2 <- grid.draw(venn.plot2)
```

![](02-Brain-RNAseqAnalysis_files/figure-gfm/unnamed-chunk-44-2.png)<!-- -->

``` r
# dev.off()

grid.newpage()
# dev.new()
# pdf("~/Desktop/Venn_GutMicrobiota_Brain_3wayComparison.pdf",height=4,width=6)  
p3 <- grid.draw(venn.plot3)
```

![](02-Brain-RNAseqAnalysis_files/figure-gfm/unnamed-chunk-44-3.png)<!-- -->

``` r
# dev.off()
```

## Save the DEG list

``` r
# write.table(AllDEGs, "~/Desktop/AllBrainDEGs.txt", sep="\t")
```

## **We find 45 genes overlapping between gut and brain DEGs. We test if this is more than expected by chance, using the full transcriptome as background**

``` r
Hyptest = phyper(44, 4988, 12374 - 4988, 91, lower.tail = FALSE)
Hyptest
```

    ## [1] 0.04770724

``` r
RepresentationFactor = 45/((91*4988)/12374)
RepresentationFactor
```

    ## [1] 1.226746

## **We test for overlap between the treatment comparisons in the brain, using the 10,493 genes retained in the differential gene expression analysis as background**

``` r
# Overlap between MD vs CL and MD vs CL_13
Hyptest = phyper(6, 25, 10493 - 25, 37, lower.tail = FALSE)
Hyptest
```

    ## [1] 1.705773e-12

``` r
RepresentationFactor = 7/((25*37)/10493)
RepresentationFactor
```

    ## [1] 79.40649

``` r
# Overlap between MD vs CL and MD vs CL_Bifi
Hyptest = phyper(9, 23, 10493 - 23, 37, lower.tail = FALSE)
Hyptest
```

    ## [1] 8.706251e-20

``` r
RepresentationFactor = 10/((23*37)/10493)
RepresentationFactor
```

    ## [1] 123.302

## **GO enrichment test with GOstats on all treatment DEGs**

``` r
#  Use GOstats to test for over-represented GO terms
#  Genes of interest vs all genes that have any GO annotation
#========================================================
library(GOstats)
library(GSEABase)
library(GOstatsPlus)

# Download GO term annotations with biomaRt
require(biomaRt)
mart <- useMart('metazoa_mart', host = 'metazoa.ensembl.org') 
mart <- useDataset('amellifera_eg_gene', mart)

annot <- getBM(
    mart = mart,
    attributes = c(
        'go_id',
        'go_linkage_type',
        'ensembl_gene_id'),
    uniqueRows = TRUE)

library(dplyr)
annot<-subset(annot, trimws(annot$go_id) !="")

goFrame=GOFrame(annot,organism="Apis mellifera")
goAllFrame=GOAllFrame(goFrame)

library("GSEABase")
gsc <- GeneSetCollection(goAllFrame, setType = GOCollection())
gene_universe = unique(unlist(geneIds(gsc)))

params <- GSEAGOHyperGParams(name="My Custom GSEA based annot Params",
                                                           geneSetCollection=gsc,
                                                           geneIds = AllDEGs,
                                                           universeGeneIds = gene_universe,
                                                           ontology = "BP",
                                                           pvalueCutoff = 0.05,
                                                           conditional = FALSE,
                                                           testDirection = "over")
BP_Over <- hyperGTest(params)

params <- GSEAGOHyperGParams(name="My Custom GSEA based annot Params",
                             geneSetCollection=gsc,
                             geneIds = AllDEGs,
                             universeGeneIds = gene_universe,
                             ontology = "MF",
                             pvalueCutoff = 0.05,
                             conditional = FALSE,
                             testDirection = "over")
MF_Over <- hyperGTest(params)

params <- GSEAGOHyperGParams(name="My Custom GSEA based annot Params",
                             geneSetCollection=gsc,
                             geneIds = AllDEGs,
                             universeGeneIds = gene_universe,
                             ontology = "CC",
                             pvalueCutoff = 0.05,
                             conditional = FALSE,
                             testDirection = "over")
CC_Over <- hyperGTest(params)
```

### **Biological process terms - hypergeometric test**

``` r
summary(BP_Over, categorySize = 10)
```

    ##       GOBPID      Pvalue OddsRatio   ExpCount Count Size
    ## 1 GO:0043094 0.002879818 31.060317 0.08242203     2   11
    ## 2 GO:0071103 0.006650372  8.737095 0.38963143     3   52
    ## 3 GO:0006323 0.007751153 17.446429 0.13487242     2   18
    ## 4 GO:0008652 0.014706685 12.119255 0.18732280     2   25
    ## 5 GO:0044281 0.024464889  2.729275 2.94471446     7  393
    ## 6 GO:0005975 0.028749737  3.761537 1.17638720     4  157
    ## 7 GO:0051276 0.031157677  3.662613 1.20635885     4  161
    ## 8 GO:0006520 0.042034468  4.193361 0.77926286     3  104
    ## 9 GO:0042592 0.046001628  6.307792 0.34467396     2   46
    ##                                       Term
    ## 1      cellular metabolic compound salvage
    ## 2                  DNA conformation change
    ## 3                            DNA packaging
    ## 4 cellular amino acid biosynthetic process
    ## 5         small molecule metabolic process
    ## 6           carbohydrate metabolic process
    ## 7                  chromosome organization
    ## 8    cellular amino acid metabolic process
    ## 9                      homeostatic process

### **Molecular function terms - hypergeometric test**

``` r
summary(MF_Over, categorySize = 10)
```

    ##       GOMFID      Pvalue OddsRatio  ExpCount Count Size
    ## 1 GO:0016769 0.005574331 20.936752 0.1140039     2   15
    ## 2 GO:0008483 0.005574331 20.936752 0.1140039     2   15
    ## 3 GO:0008061 0.007389346  8.300455 0.4028137     3   53
    ## 4 GO:0051287 0.018849095 10.446154 0.2128072     2   28
    ## 5 GO:0016616 0.043697709  6.449735 0.3344114     2   44
    ##                                                                                    Term
    ## 1                                 transferase activity, transferring nitrogenous groups
    ## 2                                                                 transaminase activity
    ## 3                                                                        chitin binding
    ## 4                                                                           NAD binding
    ## 5 oxidoreductase activity, acting on the CH-OH group of donors, NAD or NADP as acceptor

### **Cellular component terms - hypergeometric test**

``` r
summary(CC_Over, categorySize = 10)
```

    ## [1] GOCCID    Pvalue    OddsRatio ExpCount  Count     Size      Term     
    ## <0 rows> (or 0-length row.names)

# Export GO enrichment results to Excel file

``` r
library(xlsx)
write.xlsx(summary(BP_Over, categorySize = 10), file="~/Desktop/Brain_GOenrichment_GutMicrobiotaDEGs.xlsx", sheetName="BiologicalProcess", row.names=FALSE)
write.xlsx(summary(MF_Over, categorySize = 10), file="~/Desktop/Brain_GOenrichment_GutMicrobiotaDEGs.xlsx", sheetName="MolecularFunction", append=TRUE, row.names=FALSE)
# write.xlsx(summary(CC_Over, categorySize = 10), file="~/Desktop/Brain_GOenrichment_GutMicrobiotaDEGs.xlsx", sheetName="CellularComponent", append=TRUE, row.names=FALSE)
```

## **Session info**

``` r
sessionInfo()
```

    ## R version 4.1.0 (2021-05-18)
    ## Platform: x86_64-apple-darwin17.0 (64-bit)
    ## Running under: macOS Mojave 10.14.6
    ## 
    ## Matrix products: default
    ## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
    ## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
    ## 
    ## locale:
    ## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
    ## 
    ## attached base packages:
    ##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
    ##  [8] datasets  methods   base     
    ## 
    ## other attached packages:
    ##  [1] GO.db_3.13.0         dplyr_1.0.7          biomaRt_2.48.3      
    ##  [4] GOstatsPlus_0.1      GSEABase_1.54.0      annotate_1.70.0     
    ##  [7] XML_3.99-0.8         GOstats_2.58.0       graph_1.70.0        
    ## [10] Category_2.58.0      Matrix_1.3-4         AnnotationDbi_1.54.1
    ## [13] Biobase_2.52.0       VennDiagram_1.7.0    futile.logger_1.4.3 
    ## [16] ggpubr_0.4.0         xlsx_0.6.5           PCAtools_2.4.0      
    ## [19] ggrepel_0.9.1        ggplot2_3.3.5        ggplotify_0.1.0     
    ## [22] rtracklayer_1.52.1   GenomicRanges_1.44.0 GenomeInfoDb_1.28.4 
    ## [25] IRanges_2.26.0       S4Vectors_0.30.2     BiocGenerics_0.38.0 
    ## [28] edgeR_3.34.1         limma_3.48.3         data.table_1.14.2   
    ## 
    ## loaded via a namespace (and not attached):
    ##   [1] backports_1.3.0             BiocFileCache_2.0.0        
    ##   [3] plyr_1.8.6                  splines_4.1.0              
    ##   [5] BiocParallel_1.26.2         digest_0.6.29              
    ##   [7] yulab.utils_0.0.4           htmltools_0.5.2            
    ##   [9] fansi_1.0.2                 magrittr_2.0.1             
    ##  [11] memoise_2.0.0               ScaledMatrix_1.0.0         
    ##  [13] Biostrings_2.60.2           matrixStats_0.61.0         
    ##  [15] prettyunits_1.1.1           colorspace_2.0-2           
    ##  [17] rappdirs_0.3.3              blob_1.2.2                 
    ##  [19] xfun_0.28                   crayon_1.4.2               
    ##  [21] RCurl_1.98-1.5              genefilter_1.74.1          
    ##  [23] survival_3.2-13             glue_1.6.0                 
    ##  [25] gtable_0.3.0                zlibbioc_1.38.0            
    ##  [27] XVector_0.32.0              DelayedArray_0.18.0        
    ##  [29] car_3.0-12                  BiocSingular_1.8.1         
    ##  [31] Rgraphviz_2.36.0            abind_1.4-5                
    ##  [33] scales_1.1.1                futile.options_1.0.1       
    ##  [35] DBI_1.1.1                   rstatix_0.7.0              
    ##  [37] Rcpp_1.0.8                  xtable_1.8-4               
    ##  [39] progress_1.2.2              gridGraphics_0.5-1         
    ##  [41] dqrng_0.3.0                 bit_4.0.4                  
    ##  [43] rsvd_1.0.5                  AnnotationForge_1.34.1     
    ##  [45] httr_1.4.2                  ellipsis_0.3.2             
    ##  [47] pkgconfig_2.0.3             rJava_1.0-5                
    ##  [49] farver_2.1.0                dbplyr_2.1.1               
    ##  [51] locfit_1.5-9.4              utf8_1.2.2                 
    ##  [53] tidyselect_1.1.1            labeling_0.4.2             
    ##  [55] rlang_0.4.12                reshape2_1.4.4             
    ##  [57] munsell_0.5.0               tools_4.1.0                
    ##  [59] cachem_1.0.6                generics_0.1.1             
    ##  [61] RSQLite_2.2.8               broom_0.7.10               
    ##  [63] evaluate_0.14               stringr_1.4.0              
    ##  [65] fastmap_1.1.0               yaml_2.2.1                 
    ##  [67] knitr_1.36                  bit64_4.0.5                
    ##  [69] purrr_0.3.4                 KEGGREST_1.32.0            
    ##  [71] RBGL_1.68.0                 sparseMatrixStats_1.4.2    
    ##  [73] formatR_1.11                xml2_1.3.2                 
    ##  [75] compiler_4.1.0              filelock_1.0.2             
    ##  [77] curl_4.3.2                  png_0.1-7                  
    ##  [79] ggsignif_0.6.3              tibble_3.1.6               
    ##  [81] statmod_1.4.36              stringi_1.7.6              
    ##  [83] highr_0.9                   lattice_0.20-45            
    ##  [85] vctrs_0.3.8                 pillar_1.6.4               
    ##  [87] lifecycle_1.0.1             cowplot_1.1.1              
    ##  [89] bitops_1.0-7                irlba_2.3.3                
    ##  [91] R6_2.5.1                    BiocIO_1.2.0               
    ##  [93] gridExtra_2.3               lambda.r_1.2.4             
    ##  [95] assertthat_0.2.1            SummarizedExperiment_1.22.0
    ##  [97] xlsxjars_0.6.1              rjson_0.2.20               
    ##  [99] withr_2.4.3                 GenomicAlignments_1.28.0   
    ## [101] Rsamtools_2.8.0             GenomeInfoDbData_1.2.6     
    ## [103] hms_1.1.1                   beachmat_2.8.1             
    ## [105] tidyr_1.1.4                 rmarkdown_2.11             
    ## [107] DelayedMatrixStats_1.14.3   MatrixGenerics_1.4.3       
    ## [109] carData_3.0-4               restfulr_0.0.13
